diff -ru ApplePS2Controller-8/ApplePS2Controller.cpp ApplePS2Controller-8.pd1/ApplePS2Controller.cpp
--- ApplePS2Controller-8/ApplePS2Controller.cpp	2004-08-14 06:38:11.000000000 -0700
+++ ApplePS2Controller-8.pd1/ApplePS2Controller.cpp	2012-09-09 08:18:31.000000000 -0700
@@ -280,12 +280,12 @@
   //
 
   _workLoop                = IOWorkLoop::workLoop();
-  _interruptSourceMouse    = IOInterruptEventSource::interruptEventSource(
-        this, (IOInterruptEventAction) &ApplePS2Controller::interruptOccurred);
-  _interruptSourceKeyboard = IOInterruptEventSource::interruptEventSource(
-        this, (IOInterruptEventAction) &ApplePS2Controller::interruptOccurred);
-  _interruptSourceQueue    = IOInterruptEventSource::interruptEventSource(
-        this, (IOInterruptEventAction) &ApplePS2Controller::processRequestQueue);
+  _interruptSourceMouse    = IOInterruptEventSource::interruptEventSource( this,
+        OSMemberFunctionCast(IOInterruptEventAction, this, &ApplePS2Controller::interruptOccurred));
+  _interruptSourceKeyboard = IOInterruptEventSource::interruptEventSource( this,
+        OSMemberFunctionCast(IOInterruptEventAction, this, &ApplePS2Controller::interruptOccurred));
+  _interruptSourceQueue    = IOInterruptEventSource::interruptEventSource( this,
+        OSMemberFunctionCast(IOInterruptEventAction, this, &ApplePS2Controller::processRequestQueue));
 
   if ( !_workLoop                ||
        !_interruptSourceMouse    ||
