diff -ru ApplePS2Mouse-10/ApplePS2Mouse.cpp ApplePS2Mouse-10.pd1/ApplePS2Mouse.cpp
--- ApplePS2Mouse-10/ApplePS2Mouse.cpp	2004-05-13 16:17:33.000000000 -0700
+++ ApplePS2Mouse-10.pd1/ApplePS2Mouse.cpp	2012-09-09 08:30:10.000000000 -0700
@@ -136,7 +136,7 @@
   //
 
   _device->installInterruptAction(this,
-    (PS2InterruptAction)&ApplePS2Mouse::interruptOccurred);
+        OSMemberFunctionCast(PS2InterruptAction, this, &ApplePS2Mouse::interruptOccurred));
   _interruptHandlerInstalled = true;
 
   //
@@ -144,7 +144,7 @@
   //
 
   _device->installPowerControlAction( this,
-           (PS2PowerControlAction) &ApplePS2Mouse::setDevicePowerState );
+        OSMemberFunctionCast(PS2PowerControlAction, this, &ApplePS2Mouse::setDevicePowerState));
   _powerControlHandlerInstalled = true;
 
   return true;
