diff -ru AppleIntelE1000-45/AppleIntelE1000.cpp AppleIntelE1000-45.pd1/AppleIntelE1000.cpp
--- AppleIntelE1000-45/AppleIntelE1000.cpp	2012-09-24 03:17:31.000000000 +0200
+++ AppleIntelE1000-45.pd1/AppleIntelE1000.cpp	2012-10-09 02:15:13.678228337 +0200
@@ -1389,10 +1389,17 @@
 	tx_ring->size = tx_ring->count * sizeof(struct e1000_tx_desc);
 	tx_ring->size = ALIGN(tx_ring->size, 4096);
 
+#if defined(MAC_OS_X_VERSION_10_6)
 	tx_ring->pool = IOBufferMemoryDescriptor::inTaskWithPhysicalMask(kernel_task,
 						kIODirectionInOut | kIOMemoryPhysicallyContiguous, 
 						tx_ring->size,
 						0xFFFF0000UL);	// 32-bits, 64K-aligned
+#else												
+	tx_ring->pool = IOBufferMemoryDescriptor::withOptions(kIOMemoryPhysicallyContiguous, 
+						tx_ring->size,
+						PAGE_SIZE);
+#endif						
+						
 	if(tx_ring->pool){
 		tx_ring->pool->prepare();
 		tx_ring->desc = (void*)tx_ring->pool->getBytesNoCopy();
@@ -1583,10 +1590,17 @@
 
 	rx_ring->size = rx_ring->count * desc_len;
 	rx_ring->size = ALIGN(rx_ring->size, 4096);
+#if defined(MAC_OS_X_VERSION_10_6)
 
 	rx_ring->pool = IOBufferMemoryDescriptor::inTaskWithPhysicalMask(kernel_task,kIODirectionInOut | kIOMemoryPhysicallyContiguous, 
 								rx_ring->size,
 								0xFFFF0000UL);	// 32-bits, 64K-aligned
+#else							
+	rx_ring->pool = IOBufferMemoryDescriptor::withOptions(kIOMemoryPhysicallyContiguous, 
+						rx_ring->size,
+						PAGE_SIZE);							
+#endif
+								
 	if(rx_ring->pool){
 		rx_ring->pool->prepare();
 		rx_ring->desc = (void*)rx_ring->pool->getBytesNoCopy();
@@ -5306,7 +5320,7 @@
 	RELEASE(mediumDict);
 	
 	RELEASE(csrPCIAddress);
-	RELEASE(flashPCIAddress);
+	//RELEASE(flashPCIAddress);
 	
 	RELEASE(txMbufCursor);
 	
@@ -5325,7 +5339,7 @@
 	pciDevice = NULL;
 	mediumDict = NULL;
 	csrPCIAddress = NULL;
-	flashPCIAddress = NULL;
+	//flashPCIAddress = NULL;
 	interruptSource = NULL;
 	watchdogSource = NULL;
 	phyinfoSource = NULL;
@@ -5389,11 +5403,11 @@
 		E1000_DBG("csrPCIAddress.\n");
 		return false;
 	}
-	flashPCIAddress = pciDevice->mapDeviceMemoryWithRegister(kIOPCIConfigBaseAddress1);
+	/*flashPCIAddress = pciDevice->mapDeviceMemoryWithRegister(kIOPCIConfigBaseAddress1);
 	if (flashPCIAddress == NULL) {
 		E1000_DBG("flashPCIAddress.\n");
 		return false;
-	}
+	}*/
 	createWorkLoop(); // dummy call
 
 	e1000_set_mtu(ETH_DATA_LEN);
diff -ru AppleIntelE1000-45/AppleIntelE1000.h AppleIntelE1000-45.pd1/AppleIntelE1000.h
--- AppleIntelE1000-45/AppleIntelE1000.h	2012-09-24 03:17:31.000000000 +0200
+++ AppleIntelE1000-45.pd1/AppleIntelE1000.h	2012-10-09 01:52:31.000000000 +0200
@@ -121,7 +121,7 @@
     
 public:
 	IOMemoryMap * csrPCIAddress;
-	IOMemoryMap * flashPCIAddress;
+	//IOMemoryMap * flashPCIAddress;
 	
 	IOMbufNaturalMemoryCursor * txMbufCursor;
 	

