diff -ru at_cmds-54/appletalk.tproj/appletalk.c at_cmds-54.patch/appletalk.tproj/appletalk.c
--- at_cmds-54/appletalk.tproj/appletalk.c	2009-06-10 00:43:30.000000000 +0200
+++ at_cmds-54.patch/appletalk.tproj/appletalk.c	2009-06-10 00:44:53.000000000 +0200
@@ -40,6 +40,8 @@
 #include <AppleTalk/at_proto.h>
 #include <AppleTalk/at_paths.h>
 
+#define _OPEN_SOURCE_
+
 #ifndef _OPEN_SOURCE_
 #include <SystemIntegrity.h>
 #endif
Only in at_cmds-54/appletalk.tproj: appletalk.c.orig
Only in at_cmds-54/appletalk.tproj: appletalk.c.rej
