diff -ru at_cmds-54/Makefile.preamble at_cmds-54.pd1/Makefile.preamble
--- at_cmds-54/Makefile.preamble	2005-08-15 22:11:23.000000000 +0200
+++ at_cmds-54.pd1/Makefile.preamble	2012-09-14 02:50:20.570815010 +0200
@@ -20,7 +20,7 @@
 ## (e.g. change -O to -O2), see Makefile.postamble.
 
 # Flags passed to compiler (in addition to -g, -O, etc)
-OTHER_CFLAGS = -mdynamic-no-pic
+OTHER_CFLAGS = -mdynamic-no-pic -D_OPEN_SOURCE_
 # Flags passed to ld (in addition to -ObjC, etc.)
 OTHER_LDFLAGS =	
 # Flags passed to libtool when building libraries
Seulement dans at_cmds-54.pd1: Makefile.preamble~
