--- bash-106.220.2/bash-3.2/shell.c	2016-02-17 10:58:32.000000000 -0500
+++ bash-106.220.2.mod/bash-3.2/shell.c	2019-09-19 16:32:28.000000000 -0400
@@ -45,7 +45,7 @@
 
 #if defined(__APPLE__)
 #include <get_compat.h>
-#include <rootless.h>
+/* #include <rootless.h> */
 #endif /* __APPLE__ */
 
 #include "bashintl.h"
@@ -489,10 +489,10 @@
     if (-1 == setreuid(ruid, euid))
       internal_error( _("setreuid(%u,%u) failed: %s"), ruid, euid, strerror(errno));   
   }
-  int rootless = rootless_restricted_environment();
+ /* int rootless = rootless_restricted_environment();
   if (-1 == rootless)
     internal_error( _("Unable to determine rootless status: %s"), strerror(errno));
-  running_setuid |= rootless;
+  running_setuid |= rootless; */
 #else  /* !__APPLE__ */
   if (running_setuid && privileged_mode == 0)
     disable_priv_mode ();
