diff -ru bootp-170.1/IPConfiguration.bproj/ipconfigd.c bootp-170.1.pd1/IPConfiguration.bproj/ipconfigd.c
--- bootp-170.1/IPConfiguration.bproj/ipconfigd.c	2007-09-14 23:45:31.000000000 +0200
+++ bootp-170.1.pd1/IPConfiguration.bproj/ipconfigd.c	2012-09-12 19:56:40.117024302 +0200
@@ -137,6 +137,9 @@
  * - resolve the router's MAC address using ARP, and publish that
  *   information to the NetworkSignature in the IPv4 dict
  */
+ 
+ #include <CoreFoundation/CoreFoundation.h>
+ #include <CoreFoundation/CFUserNotification.h>
 
 #include <stdlib.h>
 #include <unistd.h>
