diff -ru CarbonHeaders-9G55/MacTypes.h CarbonHeaders-9G55.pd1/MacTypes.h
--- CarbonHeaders-9G55/MacTypes.h	2009-01-13 21:59:09.000000000 +0100
+++ CarbonHeaders-9G55.pd1/MacTypes.h	2012-09-09 21:07:11.502089020 +0200
@@ -348,9 +348,11 @@
   noErr                         = 0
 };
 
+#if 0
 enum {
   kNilOptions                   = 0
 };
+#endif
 
 #define kInvalidID   0
 enum {
diff -ru CarbonHeaders-9G55/Makefile CarbonHeaders-9G55.pd1/Makefile
--- CarbonHeaders-9G55/Makefile	2009-02-18 21:53:46.000000000 +0100
+++ CarbonHeaders-9G55.pd1/Makefile	2012-09-09 21:09:59.234085688 +0200
@@ -8,6 +8,7 @@
 CORESERVICES=$(DSTROOT)/System/Library/Frameworks/CoreServices.framework
 CARBONCORE=$(DSTROOT)/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/CarbonCore.framework
 CARBONCORE_HEADERS=$(CARBONCORE)/Versions/A/Headers
+CARBON_KERN_HEADERS=$(DSTROOT)/System/Library/Frameworks/Kernel.framework/Versions/A/Headers
 
 install:
 	mkdir -p $(INCLUDE)
@@ -19,6 +20,9 @@
 
 	mkdir -p $(CARBONCORE_HEADERS)
 
+	mkdir -p $(CARBON_KERN_HEADERS)
+	cp TargetConditionals.h $(CARBON_KERN_HEADERS)
+
 	ln -hfs /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers $(INCLUDE)/CarbonCore
 	ln -hfs A $(CORESERVICES)/Versions/Current
 	ln -hfs A $(CARBONCORE)/Versions/Current
Seulement dans CarbonHeaders-9G55.pd1/: Makefile~
