--- CF-476.15/CFBundle_Resources.c.origin	2008-10-27 00:22:47.000000000 +0100
+++ CF-476.15/CFBundle_Resources.c	2008-10-27 00:23:14.000000000 +0100
@@ -2173,3 +2173,11 @@
     }
     return result;
 }
+
+CFNotificationCenterRef CFNotificationCenterGetLocalCenter(void) {return;};
+void CFNotificationCenterAddObserver(CFNotificationCenterRef center, const void *observer, CFNotificationCallback callBack, CFStringRef name, const void *object, CFNotificationSuspensionBehavior suspensionBehavior){return;};
+void * const NSGenericException;
+
+CFStringRef _kCFBundleResourceSpecificationKey(CFBundleRef bundle, CFStringRef key, CFStringRef value, CFStringRef tableName) {
+	return 0;
+}
