diff -ru CFNetwork-129.20/HTTP/NTLM/ntlmBlobPriv.cpp CFNetwork-129.20.pd1/HTTP/NTLM/ntlmBlobPriv.cpp
--- CFNetwork-129.20/HTTP/NTLM/ntlmBlobPriv.cpp	2007-03-27 07:55:09.000000000 +0200
+++ CFNetwork-129.20.pd1/HTTP/NTLM/ntlmBlobPriv.cpp	2012-09-12 03:32:31.926197240 +0200
@@ -47,7 +47,9 @@
  * Private routines used by NtlmGenerator module. 
  */
 #include "ntlmBlobPriv.h"
-#include <CoreServices/CoreServices.h>
+//#include <CoreServices/CoreServices.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <CarbonCore/MacErrors.h>
 #include <sys/types.h>
 #include <sys/uio.h>
 #include <unistd.h>
@@ -455,7 +457,7 @@
 	crtn = CSSM_EncryptDataUpdate(ccHand,
 		&ptext, 1,
 		&ctext, 1,
-		&bytesEncrypted);
+		(CSSM_SIZE*)&bytesEncrypted);
 	if(crtn) {
 		#ifndef NDEBUG
 		cssmPerror("CSSM_EncryptDataUpdate", crtn);
diff -ru CFNetwork-129.20/HTTP/NTLM/ntlmBlobPriv.h CFNetwork-129.20.pd1/HTTP/NTLM/ntlmBlobPriv.h
--- CFNetwork-129.20/HTTP/NTLM/ntlmBlobPriv.h	2007-03-27 07:55:09.000000000 +0200
+++ CFNetwork-129.20.pd1/HTTP/NTLM/ntlmBlobPriv.h	2012-09-12 03:24:06.514207280 +0200
@@ -55,7 +55,8 @@
 #include <stdint.h>
 #include <Security/cssmtype.h>
 #include <Security/SecBase.h>
-#include <CoreServices/CoreServices.h>
+//#include <CoreServices/CoreServices.h>
+#include <CoreFoundation/CoreFoundation.h>
 
 #ifdef  __cplusplus
 extern "C" {
diff -ru CFNetwork-129.20/HTTP/NTLM/NtlmGenerator.cpp CFNetwork-129.20.pd1/HTTP/NTLM/NtlmGenerator.cpp
--- CFNetwork-129.20/HTTP/NTLM/NtlmGenerator.cpp	2007-03-27 07:55:09.000000000 +0200
+++ CFNetwork-129.20.pd1/HTTP/NTLM/NtlmGenerator.cpp	2012-09-12 03:28:08.222202478 +0200
@@ -55,7 +55,9 @@
 
 #include "NtlmGenerator.h"
 #include "ntlmBlobPriv.h"
-#include <CoreServices/CoreServices.h>
+//#include <CoreServices/CoreServices.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <CarbonCore/MacErrors.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
