--- configd-212.2/SystemConfiguration.fproj/SystemConfiguration.h.origin	2008-11-12 15:52:25.000000000 +0100
+++ configd-212.2/SystemConfiguration.fproj/SystemConfiguration.h	2008-11-12 15:53:08.000000000 +0100
@@ -109,6 +109,8 @@
 #include <SystemConfiguration/SCDynamicStoreKey.h>
 #include <SystemConfiguration/SCDynamicStoreCopySpecific.h>
 
+#include <CarbonCore/MacTypes.h>
+
 /* SCPreferences APIs */
 #include <SystemConfiguration/SCPreferences.h>
 #include <SystemConfiguration/SCPreferencesPath.h>
