--- configd-212.2/SystemConfiguration.fproj/SCDPrivate.c.origin	2008-10-30 01:47:41.000000000 +0100
+++ configd-212.2/SystemConfiguration.fproj/SCDPrivate.c	2008-10-30 01:49:05.000000000 +0100
@@ -987,10 +987,10 @@
 					     set);
 	newStr = CFStringCreateMutableCopy(NULL, 0, str);
 	CFStringNormalize(newStr, kCFStringNormalizationFormC);
-	if (!CFStringTransform(newStr, NULL, transform, FALSE)) {
+	/*if (!CFStringTransform(newStr, NULL, transform, FALSE)) {
 		CFRelease(newStr);
 		newStr = NULL;
-	}
+	}*/
 	CFRelease(transform);
 
     done :
