diff -ru configd-212.2/scutil.tproj/prefs.c configd-212.2.pd1/scutil.tproj/prefs.c
--- configd-212.2/scutil.tproj/prefs.c	2008-03-13 02:39:52.000000000 +0100
+++ configd-212.2.pd1/scutil.tproj/prefs.c	2012-09-10 03:32:50.053629393 +0200
@@ -34,6 +34,8 @@
 #include <sys/stat.h>
 #include <dlfcn.h>
 
+#include <CarbonCore/MacTypes.h>
+
 #include <SystemConfiguration/SCPreferencesSetSpecific.h>
 #include <Security/Authorization.h>
 
