--- configd-212.2/SystemConfiguration.fproj/SCNetworkConnectionPrivate.c.origin	2008-10-30 01:44:55.000000000 +0100
+++ configd-212.2/SystemConfiguration.fproj/SCNetworkConnectionPrivate.c	2008-10-30 01:46:56.000000000 +0100
@@ -207,9 +207,9 @@
 {
 	Boolean	ok;
 
-	if (CFPreferencesAppValueIsForced(serviceID, USER_PREFERENCES_APPLICATION_ID)) {
+	/*if (CFPreferencesAppValueIsForced(serviceID, USER_PREFERENCES_APPLICATION_ID)) {
 		return FALSE;
-	}
+	}*/
 
 	CFPreferencesSetValue(serviceID,
 			      newPreferences,
@@ -352,7 +352,8 @@
 		return FALSE;
 	}
 
-	return CFPreferencesAppValueIsForced(userPrivate->serviceID, USER_PREFERENCES_APPLICATION_ID);
+	//return CFPreferencesAppValueIsForced(userPrivate->serviceID, USER_PREFERENCES_APPLICATION_ID);
+	return;
 }
 
 
