--- configd-212.2/SystemConfiguration.fproj/SCPrivate.h.origin	2008-11-12 15:49:23.000000000 +0100
+++ configd-212.2/SystemConfiguration.fproj/SCPrivate.h	2008-11-12 15:50:12.000000000 +0100
@@ -36,6 +36,8 @@
 #include <SystemConfiguration/SCDynamicStoreCopySpecificPrivate.h>
 #include <SystemConfiguration/SCDynamicStoreSetSpecificPrivate.h>
 
+#include <CarbonCore/MacTypes.h>
+
 /* SCPreferences SPIs */
 #include <SystemConfiguration/SCPreferencesPrivate.h>
 #include <SystemConfiguration/SCPreferencesGetSpecificPrivate.h>
