diff -ru DirectoryService-514.25/PlugIns/Common/BaseDirectoryPlugin.cpp DirectoryService-514.25.pd1/PlugIns/Common/BaseDirectoryPlugin.cpp
--- DirectoryService-514.25/PlugIns/Common/BaseDirectoryPlugin.cpp	2009-02-05 00:33:24.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/Common/BaseDirectoryPlugin.cpp	2012-09-11 20:24:36.710707255 +0200
@@ -37,7 +37,7 @@
 
 extern "C" int ConvertXMLPolicyToSpaceDelimited( const char *inXMLDataStr, char **outPolicyStr );
 
-#ifndef __OBJC__
+#if 1
 	#define EXCEPTION_START		try {
 	#define EXCEPTION_END		} catch ( ... ) { \
 									DbgLog( kLogPlugin, "%s: uncaught exception in %s:%d", fPlugInName, __FILE__, __LINE__ ); \
@@ -561,7 +561,7 @@
 		goto failure;
 	}
 
-#ifndef __OBJC__
+#if 1
 	cfNodeName = pContext->fVirtualNode->CopyNodeName();
 #else
 	cfNodeName = (CFStringRef) [pContext->fVirtualNode copyNodeName];
@@ -699,7 +699,7 @@
 																		   inData->fInRequestData->fBufferLength, kCFAllocatorNull );
 		CFMutableDataRef	cfResponseData	= CFDataCreateMutable( kCFAllocatorDefault, inData->fOutRequestResponse->fBufferSize );
 		
-#ifndef __OBJC__
+#if 1
 		siResult = pContext->fVirtualNode->CustomCall( inData->fInRequestCode, cfRequestData, cfResponseData, 
 													   inData->fOutRequestResponse->fBufferSize );
 #else
@@ -760,7 +760,7 @@
 				pContext = (sBDPINodeContext *) MakeContextData( kBDPIDirNode );
 				if ( pContext == NULL )
 				{
-#ifndef __OBJC__
+#if 1
 					delete pEngine;
 #else
 					[pEngine release];
@@ -885,7 +885,7 @@
 		
 		if ( cfAttributes != NULL )
 		{
-#ifndef __OBJC__
+#if 1
 			cfNodeInfo = pContext->fVirtualNode->CopyNodeInfo( cfAttributes );
 #else
 			cfNodeInfo = (CFMutableDictionaryRef) [pContext->fVirtualNode copyNodeInfo: (NSArray *)cfAttributes];
@@ -1054,7 +1054,7 @@
 	if( pUsername != NULL && CFStringGetLength(pUsername) != 0 )
 	{
 		// we don't use name and password yet...
-#ifndef __OBJC__
+#if 1
 		CFDictionaryRef	userPolicy = inContext->fVirtualNode->CopyPasswordPolicyForRecord( inRecordType, pUsername );
 #else
 		CFDictionaryRef	userPolicy = (CFDictionaryRef) [inContext->fVirtualNode copyPasswordPolicyForRecord: (NSString *) inRecordType withName: (NSString *)pUsername];
@@ -1153,7 +1153,7 @@
 	
 	cfPassword = CFStringCreateWithBytes( kCFAllocatorDefault, (const UInt8 *) pData, stringLen, kCFStringEncodingUTF8, false );
 
-#ifndef __OBJC__
+#if 1
 	if ( bAuthOnly )
 		siResult = inContext->fVirtualNode->VerifyCredentials( inRecordType, cfUserName, cfPassword );
 	else
@@ -1256,7 +1256,7 @@
 		cfPassword2 = CFStringCreateWithBytes( kCFAllocatorDefault, (const UInt8 *) pData, stringLen, kCFStringEncodingUTF8, false );
 	}
 
-#ifndef __OBJC__
+#if 1
 	siResult = inContext->fVirtualNode->ChangePassword( inRecordType, cfUserName, (cfPassword2 ? cfPassword1 : NULL), (cfPassword2 ? cfPassword2 : cfPassword1) );
 #else
 	siResult = [inContext->fVirtualNode changePassword: (NSString *)inRecordType recordName: (NSString *)cfUserName oldPassword: (NSString *)(cfPassword2 ? cfPassword1 : NULL) 
@@ -1341,7 +1341,7 @@
 	
 	inData->fOutRecEntryCount = 0;
 
-#ifndef __OBJC__
+#if 1
 	siResult = pContext->fVirtualNode->SearchRecords( pContinue, inData->fInDataBuff, &(inData->fOutRecEntryCount) );
 #else
 	siResult = [pContext->fVirtualNode searchRecords: pContinue buffer: inData->fInDataBuff outCount: &(inData->fOutRecEntryCount)];
@@ -1483,7 +1483,7 @@
 	
 	inData->fOutMatchRecordCount = 0;
 	
-#ifndef __OBJC__
+#if 1
 	siResult = pContext->fVirtualNode->SearchRecords( pContinue, inData->fOutDataBuff, &(inData->fOutMatchRecordCount) );
 #else
 	siResult = [pContext->fVirtualNode searchRecords: pContinue buffer: inData->fOutDataBuff outCount: &(inData->fOutMatchRecordCount)];
@@ -1555,7 +1555,7 @@
 	cfRecName = CFStringCreateWithCString( kCFAllocatorDefault, inData->fInRecName->fBufferData, kCFStringEncodingUTF8 );
 	cfRecType = CFStringCreateWithCString( kCFAllocatorDefault, inData->fInRecType->fBufferData, kCFStringEncodingUTF8 );
 
-#ifndef __OBJC__
+#if 1
 	pRecord = pContext->fVirtualNode->RecordOpen( cfRecType, cfRecName );
 #else
 	pRecord = (CFMutableDictionaryRef) [pContext->fVirtualNode recordOpenName: (NSString *)cfRecName recordType: (NSString *)cfRecType];
@@ -1735,7 +1735,7 @@
 	// arbitrary max length
 	cfRecType = (CFStringRef) CFDictionaryGetValue( pContext->fRecord, kBDPITypeKey );
 
-#ifndef __OBJC__
+#if 1
 	pOutAttrEntry->fAttributeValueMaxSize = pContext->fVirtualNode->MaximumSizeForAttribute( cfRecType, cfAttributeName );
 #else
 	pOutAttrEntry->fAttributeValueMaxSize = [pContext->fVirtualNode maximumSize: (NSString *)cfRecType forAttribute: (NSString *)cfAttributeName];
@@ -1778,7 +1778,7 @@
 	cfRecName = CFStringCreateWithCString( kCFAllocatorDefault, inData->fInRecName->fBufferData, kCFStringEncodingUTF8 );
 	cfRecType = CFStringCreateWithCString( kCFAllocatorDefault, inData->fInRecType->fBufferData, kCFStringEncodingUTF8 );
 
-#ifndef __OBJC__
+#if 1
 	siResult = pContext->fVirtualNode->RecordCreate( cfRecType, cfRecName );
 #else
 	siResult = [pContext->fVirtualNode recordCreateName: (NSString *)cfRecName recordType: (NSString *)cfRecType];
@@ -1786,7 +1786,7 @@
 
 	if( siResult == eDSNoErr && (inData->fType == kCreateRecordAndOpen || inData->fInOpen) )
 	{
-#ifndef __OBJC__
+#if 1
 		CFMutableDictionaryRef	pRecord = pContext->fVirtualNode->RecordOpen( cfRecType, cfRecName );
 #else
 		CFMutableDictionaryRef	pRecord = (CFMutableDictionaryRef) [pContext->fVirtualNode recordOpenName: (NSString *)cfRecName recordType: (NSString *)cfRecType];
@@ -1825,7 +1825,7 @@
 	sBDPIRecordEntryContext *pContext = (sBDPIRecordEntryContext *) fContextHash->GetItemData( inData->fInRecRef );
 	if ( pContext != NULL )
 	{
-#ifndef __OBJC__
+#if 1
 		if ( pContext->fVirtualNode->AllowChangesForAttribute(pContext->fRecord, CFSTR(kDSNAttrRecordType)) )
 #else
 		if ( [pContext->fVirtualNode allowChanges: (NSDictionary *)pContext->fRecord forAttribute: @kDSNAttrRecordType] )
@@ -1834,7 +1834,7 @@
 			CFStringRef	cfType = CFStringCreateWithCString( kCFAllocatorDefault, inData->fInNewRecType->fBufferData, kCFStringEncodingUTF8 );
 			if ( cfType != NULL )
 			{
-#ifndef __OBJC__
+#if 1
 				siResult = (tDirStatus) pContext->fVirtualNode->RecordSetType( pContext->fRecord, cfType );
 #else
 				siResult = (tDirStatus) [pContext->fVirtualNode record: (NSMutableDictionary *)pContext->fRecord setRecordType: (NSString *)cfType];
@@ -1867,7 +1867,7 @@
 	sBDPIRecordEntryContext *pContext = (sBDPIRecordEntryContext *) fContextHash->GetItemData( inData->fInRecRef );
 	if ( pContext != NULL )
 	{
-#ifndef __OBJC__
+#if 1
 		siResult = (tDirStatus)pContext->fVirtualNode->RecordDelete( pContext->fRecord );
 #else
 		siResult = (tDirStatus)[pContext->fVirtualNode recordDelete: (NSDictionary *)pContext->fRecord];
@@ -1890,7 +1890,7 @@
 	sBDPIRecordEntryContext *pContext = (sBDPIRecordEntryContext *) fContextHash->GetItemData( inData->fInRecRef );
 	if ( pContext != NULL )
 	{
-#ifndef __OBJC__
+#if 1
 		if ( pContext->fVirtualNode->AllowChangesForAttribute(pContext->fRecord, CFSTR(kDSNAttrRecordName)) )
 #else
 		if ( [pContext->fVirtualNode allowChanges: (NSDictionary *)pContext->fRecord forAttribute: @kDSNAttrRecordName] )
@@ -1903,7 +1903,7 @@
 				CFArrayRef cfValues = CFArrayCreate( kCFAllocatorDefault, (const void **) &cfValue, 1, &kCFTypeArrayCallBacks );
 				if ( cfValues != NULL )
 				{
-#ifndef __OBJC__
+#if 1
 					siResult = pContext->fVirtualNode->RecordSetValuesForAttribute( pContext->fRecord, CFSTR(kDSNAttrRecordName), cfValues );
 #else
 					siResult = [pContext->fVirtualNode record: (NSMutableDictionary *)pContext->fRecord setValues: (NSArray *) cfValues 
@@ -1941,7 +1941,7 @@
 	sBDPIRecordEntryContext *pContext = (sBDPIRecordEntryContext *) fContextHash->GetItemData( inData->fInRecRef );
 	if ( pContext != NULL )
 	{
-#ifndef __OBJC__
+#if 1
 		pContext->fVirtualNode->RecordFlush( pContext->fRecord );
 #else
 		[pContext->fVirtualNode recordFlush: (NSDictionary *)pContext->fRecord];
@@ -2043,7 +2043,7 @@
 
 					pAttribInfo->fAttributeValueCount				= valueCount;
 					pAttribInfo->fAttributeDataSize					= uiValueSize;
-#ifndef __OBJC__
+#if 1
 					pAttribInfo->fAttributeValueMaxSize				= pRecordContext->fVirtualNode->MaximumSizeForAttribute( cfRecType, cfKey );
 #else
 					pAttribInfo->fAttributeValueMaxSize				= [pRecordContext->fVirtualNode maximumSize: (NSString *)cfRecType 
@@ -2482,7 +2482,7 @@
 		
 		if ( cfNewAttribute != NULL )
 		{
-#ifndef __OBJC__
+#if 1
 			if ( pContext->fVirtualNode->AllowChangesForAttribute(pContext->fRecord, cfNewAttribute) )
 #else
 			if ( [pContext->fVirtualNode allowChanges: (NSDictionary *)pContext->fRecord forAttribute: (NSString *)cfNewAttribute] )
@@ -2522,7 +2522,7 @@
 
 					if ( cfNewValue != NULL )
 					{
-#ifndef __OBJC__
+#if 1
 						siResult = pContext->fVirtualNode->RecordSetValuesForAttribute( pContext->fRecord, cfNewAttribute, cfNewValue );
 #else
 						siResult = [pContext->fVirtualNode record: (NSMutableDictionary *)pContext->fRecord 
@@ -2568,7 +2568,7 @@
 		
 		if ( cfAttribName != NULL )
 		{
-#ifndef __OBJC__
+#if 1
 			if ( pContext->fVirtualNode->AllowChangesForAttribute(pContext->fRecord, cfAttribName) )
 #else
 			if ( [pContext->fVirtualNode allowChanges: (NSDictionary *)pContext->fRecord forAttribute: (NSString *)cfAttribName] )
@@ -2577,7 +2577,7 @@
 				CFMutableArrayRef	cfEmptyArray = CFArrayCreateMutable( kCFAllocatorDefault, 0, &kCFTypeArrayCallBacks );
 				
 				if ( cfEmptyArray != NULL )
-#ifndef __OBJC__
+#if 1
 					siResult = pContext->fVirtualNode->RecordSetValuesForAttribute( pContext->fRecord, cfAttribName, cfEmptyArray );
 #else
 					siResult = [pContext->fVirtualNode record: (NSMutableDictionary *)pContext->fRecord setValues: (NSArray *)cfEmptyArray
@@ -2620,7 +2620,7 @@
 		
 		if ( cfAttribName != NULL )
 		{
-#ifndef __OBJC__
+#if 1
 			if ( pContext->fVirtualNode->AllowChangesForAttribute(pContext->fRecord, cfAttribName) )
 #else
 			if ( [pContext->fVirtualNode allowChanges: (NSDictionary *)pContext->fRecord forAttribute:(NSString *)cfAttribName] )
@@ -2657,7 +2657,7 @@
 					
 					if ( cfNewValues != NULL )
 					{
-#ifndef __OBJC__
+#if 1
 						siResult = pContext->fVirtualNode->RecordAddValuesToAttribute( pContext->fRecord, cfAttribName, cfNewValues );
 #else
 						siResult = [pContext->fVirtualNode record: (NSMutableDictionary *)pContext->fRecord 
@@ -2732,7 +2732,7 @@
 					
 					if ( CalcCRCWithLength(pValue, valueLen) == inData->fInAttrValueID )
 					{
-#ifndef __OBJC__
+#if 1
 						siResult = pContext->fVirtualNode->RecordRemoveValueFromAttribute( pContext->fRecord, cfAttribute, cfValue );
 #else
 						siResult = [pContext->fVirtualNode record: (NSMutableDictionary *)pContext->fRecord
@@ -2778,7 +2778,7 @@
 		goto failure;
 	}
 	
-#ifndef __OBJC__
+#if 1
 	if ( pContext->fVirtualNode->AllowChangesForAttribute(pContext->fRecord, cfAttribute) )
 #else
 	if ( [pContext->fVirtualNode allowChanges: (NSDictionary *)pContext->fRecord forAttribute: (NSString *)cfAttribute] )
@@ -2870,7 +2870,7 @@
 			cfValues = (CFMutableArrayRef) CreateCFArrayFromList( ((sSetAttributeValues *)inData)->fInAttrValueList );
 		}
 		
-#ifndef __OBJC__
+#if 1
 		if ( cfValues != NULL )
 			siResult = pContext->fVirtualNode->RecordSetValuesForAttribute( pContext->fRecord, cfAttribute, cfValues );
 		else
@@ -3117,7 +3117,7 @@
 	{
         case kBDPIDirNode:
 			tmpNode = (sBDPINodeContext *) inContext;
-#ifndef __OBJC__
+#if 1
 			DSDelete( tmpNode->fVirtualNode );
 #else
 			[tmpNode->fVirtualNode release];
diff -ru DirectoryService-514.25/PlugIns/Common/BaseDirectoryPluginTypes.h DirectoryService-514.25.pd1/PlugIns/Common/BaseDirectoryPluginTypes.h
--- DirectoryService-514.25/PlugIns/Common/BaseDirectoryPluginTypes.h	2007-02-07 08:25:43.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/Common/BaseDirectoryPluginTypes.h	2012-09-11 20:24:40.770707174 +0200
@@ -36,7 +36,7 @@
 #define kBDPITypeKey			CFSTR("Type")
 #define kBDPIGUID				CFSTR("GUID")
 
-#ifdef __OBJC__
+#if 0
 	#define kNSBDPINameKey			(NSString*)kBDPINameKey
 	#define kNSBDPIDistinguishedKey	(NSString*)kBDPIDistinguishedKey
 	#define kNSBDPIAttributeKey		(NSString*)kBDPIAttributeKey
@@ -44,7 +44,7 @@
 	#define kNSBDPIGUID				(NSString*)kBDPIGUID
 #endif
 
-#ifndef __OBJC__
+#if 1
 class BDPIVirtualNode;
 #else
 @class BDPIVirtualNode;
diff -ru DirectoryService-514.25/PlugIns/Common/BDPIVirtualNode.cpp DirectoryService-514.25.pd1/PlugIns/Common/BDPIVirtualNode.cpp
--- DirectoryService-514.25/PlugIns/Common/BDPIVirtualNode.cpp	2007-02-07 08:25:43.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/Common/BDPIVirtualNode.cpp	2012-09-11 20:24:38.722707215 +0200
@@ -24,7 +24,7 @@
 #include "BDPIVirtualNode.h"
 #include "BaseDirectoryPlugin.h"
 
-#ifdef __OBJC__
+#if 0
 
 @implementation BDPIVirtualNode
 
diff -ru DirectoryService-514.25/PlugIns/Common/BDPIVirtualNode.h DirectoryService-514.25.pd1/PlugIns/Common/BDPIVirtualNode.h
--- DirectoryService-514.25/PlugIns/Common/BDPIVirtualNode.h	2007-02-07 16:25:34.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/Common/BDPIVirtualNode.h	2012-09-11 20:24:42.562707138 +0200
@@ -26,7 +26,7 @@
 
 #include <DirectoryServiceCore/BaseDirectoryPluginTypes.h>
 
-#ifndef __OBJC__
+#if 1
 
 // Virtual class for BaseDirectoryPlugInVirtualNode (BDPIVirtualNode)
 class BDPIVirtualNode
diff -ru DirectoryService-514.25/PlugIns/LDAPv3/CLDAPv3Plugin.h DirectoryService-514.25.pd1/PlugIns/LDAPv3/CLDAPv3Plugin.h
--- DirectoryService-514.25/PlugIns/LDAPv3/CLDAPv3Plugin.h	2008-03-14 18:03:23.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/LDAPv3/CLDAPv3Plugin.h	2012-09-11 20:24:35.130707286 +0200
@@ -308,7 +308,7 @@
 		virtual bool			CheckConfiguration		( const char *inData, UInt32 inLength );
 		virtual tDirStatus		HandleCustomCall		( sBDPINodeContext *pContext, sDoPlugInCustomCall *inData );
 		virtual bool			IsConfigureNodeName		( CFStringRef inNodeName );
-#ifndef __OBJC__
+#if 1
 		virtual BDPIVirtualNode	*CreateNodeForPath		( CFStringRef inPath, uid_t inUID, uid_t inEffectiveUID );
 #else
 		virtual id<BDPIVirtualNode>	CreateNodeForPath	( CFStringRef inPath, uid_t inUID, uid_t inEffectiveUID );
