diff -ru DirectoryService-514.25/PlugIns/BSD/FlatFileNode.h DirectoryService-514.25.pd1/PlugIns/BSD/FlatFileNode.h
--- DirectoryService-514.25/PlugIns/BSD/FlatFileNode.h	2008-03-14 16:09:41.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/BSD/FlatFileNode.h	2012-09-12 13:30:26.000000000 +0200
@@ -24,6 +24,7 @@
 #ifndef _FLATFILENODE_H
 #define _FLATFILENODE_H
 
+#include <CoreFoundation/CFFileDescriptor.h>
 #include <sqlite3.h>
 #include <map>
 #include <string>		//STL string class
diff -ru DirectoryService-514.25/PlugIns/LDAPv3/DSAPIWrapper.cpp DirectoryService-514.25.pd1/PlugIns/LDAPv3/DSAPIWrapper.cpp
--- DirectoryService-514.25/PlugIns/LDAPv3/DSAPIWrapper.cpp	2008-04-29 19:46:21.000000000 +0200
+++ DirectoryService-514.25.pd1/PlugIns/LDAPv3/DSAPIWrapper.cpp	2012-09-12 13:31:38.000000000 +0200
@@ -90,7 +90,7 @@
 DSAPIWrapper::OpenSpecificPasswordServerNode( const char *inServerAddress )
 {
 	char pwServerNodeStr[256];
-    tDataList *pDataList = nil;
+    tDataList *pDataList = NULL;
 	tDirStatus status = eDSNoErr;
 	
 	if ( inServerAddress == NULL )
diff -ru DirectoryService-514.25/PlugIns/Local/CDSLocalPluginNode.h DirectoryService-514.25.pd1/PlugIns/Local/CDSLocalPluginNode.h
--- DirectoryService-514.25/PlugIns/Local/CDSLocalPluginNode.h	2008-03-14 16:09:41.000000000 +0100
+++ DirectoryService-514.25.pd1/PlugIns/Local/CDSLocalPluginNode.h	2012-09-12 13:27:48.000000000 +0200
@@ -29,6 +29,7 @@
 #ifndef _CDSLocalPluginNode_
 #define _CDSLocalPluginNode_	1
 
+#include <CoreFoundation/CFFileDescriptor.h>
 #include "CDSLocalPlugin.h"
 #include "DSMutexSemaphore.h"
 #include "DSEventSemaphore.h"
