diff -ru DirectoryService-514.25/Threads/DSMutexSemaphore.cpp DirectoryService-514.25.pd1/Threads/DSMutexSemaphore.cpp
--- DirectoryService-514.25/Threads/DSMutexSemaphore.cpp	2007-06-08 22:54:57.000000000 +0200
+++ DirectoryService-514.25.pd1/Threads/DSMutexSemaphore.cpp	2012-09-11 04:36:27.871837299 +0200
@@ -37,7 +37,7 @@
 #include <sys/types.h>
 #include <sys/sysctl.h>
 #include <unistd.h>		// for _POSIX_THREADS
-#include "dslockstat.h"
+//#include "dslockstat.h"
 
 // Uncomment the following line in addition to enabling DEBUG_LOCKS and we will use try lock to acquire a lock and spin
 // until we do.  After 60 seconds we will log what locks we are waiting on
@@ -159,8 +159,8 @@
 	}
 	else
 	{
-		if ( mLockHistoryInfo->fShouldDTrace )
-			DSLOCKSTAT_MUTEX_RELEASE( (long) &mMutex, (char *) mMutexName, "no file info", 0 );
+		/*if ( mLockHistoryInfo->fShouldDTrace )
+			DSLOCKSTAT_MUTEX_RELEASE( (long) &mMutex, (char *) mMutexName, "no file info", 0 );*/
 	}
 }
 
@@ -267,7 +267,7 @@
     int error = pthread_mutex_unlock( &mMutex );
 	if ( error == 0 )
 	{
-		if ( DSLOCKSTAT_MUTEX_RELEASE_ENABLED() && mLockHistoryInfo->fShouldDTrace )
+		/*if ( DSLOCKSTAT_MUTEX_RELEASE_ENABLED() && mLockHistoryInfo->fShouldDTrace )
 		{
 			char *shortName = rindex( file, '/' );
 			if ( shortName == NULL )
@@ -275,7 +275,7 @@
 			else 
 				shortName++;
 			DSLOCKSTAT_MUTEX_RELEASE( (long) &mMutex, (char *) mMutexName, shortName, line );
-		}
+		}*/
 		
 #if defined(DEBUG_LOCKS)
 		if ( (--mLockHistoryInfo->fThreadOwnerCount) == 0 )
@@ -336,7 +336,7 @@
 	if ( error == 0 )
 	{
 		// we have our own flag so we don't dtrace certain locks
-		if ( DSLOCKSTAT_MUTEX_ACQUIRE_ENABLED() && mLockHistoryInfo->fShouldDTrace )
+		/*if ( DSLOCKSTAT_MUTEX_ACQUIRE_ENABLED() && mLockHistoryInfo->fShouldDTrace )
 		{
 			char *shortName = rindex( file, '/' );
 			if ( shortName == NULL )
@@ -344,7 +344,7 @@
 			else 
 				shortName++;
 			DSLOCKSTAT_MUTEX_ACQUIRE( (long) &mMutex, (char *) mMutexName, shortName, line );
-		}
+		}*/
 
 #if defined(DEBUG_LOCKS)
 		OSSpinLockLock( &mLockHistoryInfo->fOSLock );
@@ -383,7 +383,7 @@
 	int error = pthread_mutex_trylock( &mMutex );
 	if ( error == 0 )
 	{
-		if ( DSLOCKSTAT_MUTEX_ACQUIRE_ENABLED() && mLockHistoryInfo->fShouldDTrace )
+		/*if ( DSLOCKSTAT_MUTEX_ACQUIRE_ENABLED() && mLockHistoryInfo->fShouldDTrace )
 		{
 			char *shortName = rindex( file, '/' );
 			if ( shortName == NULL )
@@ -391,7 +391,7 @@
 			else 
 				shortName++;
 			DSLOCKSTAT_MUTEX_ACQUIRE( (long) &mMutex, (char *) mMutexName, shortName, line );
-		}
+		}*/
 		
 #if defined(DEBUG_LOCKS)
 		OSSpinLockLock( &mLockHistoryInfo->fOSLock );
