diff -ru DSPasswordServerPlugin-208.6/AuthFile.mm DSPasswordServerPlugin-208.6.pd1/AuthFile.mm
--- DSPasswordServerPlugin-208.6/AuthFile.mm	2007-03-12 23:40:05.000000000 +0100
+++ DSPasswordServerPlugin-208.6.pd1/AuthFile.mm	2012-09-11 05:27:20.211776665 +0200
@@ -38,7 +38,9 @@
 #endif
 
 #include <TargetConditionals.h>
-#include <CoreServices/CoreServices.h>
+//#include <CoreServices/CoreServices.h>
+#include <CarbonCore/endian.h>
+
 #include <CommonCrypto/CommonDigest.h>
 
 #include <sys/param.h>
Seulement dans DSPasswordServerPlugin-208.6.pd1/: AuthFile.mm~
diff -ru DSPasswordServerPlugin-208.6/CAuthFileBase.cpp DSPasswordServerPlugin-208.6.pd1/CAuthFileBase.cpp
--- DSPasswordServerPlugin-208.6/CAuthFileBase.cpp	2007-01-11 02:30:55.000000000 +0100
+++ DSPasswordServerPlugin-208.6.pd1/CAuthFileBase.cpp	2012-09-11 05:03:49.887804682 +0200
@@ -31,7 +31,7 @@
 #include <unistd.h>
 #include <openssl/err.h>
 
-#include <CoreServices/CoreServices.h>
+//#include <CoreServices/CoreServices.h>
 #include <TargetConditionals.h>
 
 #include "DSMutexSemaphore.h"
diff -ru DSPasswordServerPlugin-208.6/CPSUtilities.cpp DSPasswordServerPlugin-208.6.pd1/CPSUtilities.cpp
--- DSPasswordServerPlugin-208.6/CPSUtilities.cpp	2007-09-14 00:57:15.000000000 +0200
+++ DSPasswordServerPlugin-208.6.pd1/CPSUtilities.cpp	2012-09-11 05:05:44.151802411 +0200
@@ -59,6 +59,9 @@
 #include "SASLCode.h"
 #include "CAuthFileBase.h"
 
+#include <CarbonCore/MacTypes.h>
+
+
 #define kMinTrialTime		140000
 #define kMaxTrialTime		1250000
 #define kMaxIPAddrs			32
diff -ru DSPasswordServerPlugin-208.6/CReplicaFile.cpp DSPasswordServerPlugin-208.6.pd1/CReplicaFile.cpp
--- DSPasswordServerPlugin-208.6/CReplicaFile.cpp	2008-04-15 23:28:49.000000000 +0200
+++ DSPasswordServerPlugin-208.6.pd1/CReplicaFile.cpp	2012-09-11 05:02:09.791806669 +0200
@@ -1871,7 +1871,7 @@
 	char *tptr = outHexStr;
 	char base16table[16] = { '0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F' };
 	
-    if ( inData == nil || outHexStr == nil )
+    if ( inData == NULL || outHexStr == NULL )
         return false;
     
 	for ( int idx = 0; idx < len; idx++ )
diff -ru DSPasswordServerPlugin-208.6/SMBAuth.c DSPasswordServerPlugin-208.6.pd1/SMBAuth.c
--- DSPasswordServerPlugin-208.6/SMBAuth.c	2007-02-07 01:06:59.000000000 +0100
+++ DSPasswordServerPlugin-208.6.pd1/SMBAuth.c	2012-09-11 05:31:21.999771863 +0200
@@ -31,7 +31,11 @@
 #include <syslog.h>
 #include <CommonCrypto/CommonCryptor.h>
 
-#include <CoreServices/CoreServices.h>
+//#include <CoreServices/CoreServices.h>
+#include <CarbonCore/Endian.h>
+#include <CoreFoundation/CoreFoundation.h>
+
+
 
 /*!
  * @header SMBAuth
