--- dtrace-48/libproc/libproc_apple.h.origin	2008-11-09 16:43:03.000000000 +0100
+++ dtrace-48/libproc/libproc_apple.h	2008-11-09 16:43:24.000000000 +0100
@@ -45,9 +45,9 @@
 struct ps_prochandle {
 	task_t task;
 	pstatus_t status;
-	VMUSymbolicator* symbolicator;
-	VMUSymbolicator* prev_symbolicator;
-	NSMutableDictionary* prmap_dictionary;
+	void** symbolicator; //VMUSymbolicator* symbolicator;
+	void** prev_symbolicator; //VMUSymbolicator* prev_symbolicator;
+	void** prmap_dictionary; //NSMutableDictionary* prmap_dictionary;
 	mach_port_t dtrace_dyld_port;
 	Phandler_func_t *activity_handler_func;
 	void *activity_handler_arg;
