--- dtrace-48/libdtrace/dt_pid_apple.m.origin	2008-11-09 16:21:22.000000000 +0100
+++ dtrace-48/libdtrace/dt_pid_apple.m	2008-11-09 16:23:08.000000000 +0100
@@ -1,7 +1,8 @@
 #if defined(__APPLE__)
 
-#import <Symbolication/Symbolication.h>
-#import <Symbolication/SymbolicationPrivate.h>
+// Symbolication is disabled because it doesn't belong to Darwin and is a blocker for PureDarwin
+//#import <Symbolication/Symbolication.h>
+//#import <Symbolication/SymbolicationPrivate.h>
 
 // This must be done *after* any references to Foundation.h!
 #define uint_t  __Solaris_uint_t
@@ -118,4 +119,4 @@
 	return (ret);
 }
 
-#endif /* __APPLE__ */
\ No newline at end of file
+#endif /* __APPLE__ */
