diff --git a/lib/libproc/libproc.c b/lib/libproc/libproc.c
index e934cfe..70428b7 100644
--- a/lib/libproc/libproc.c
+++ b/lib/libproc/libproc.c
@@ -495,30 +495,6 @@ update_check_symbol_gen(struct ps_prochandle *P, CSSymbolRef symbol, bool err_on
 }
 #endif /* DTRACE_USE_CORESYMBOLICATION */
 
-#if defined(__arm__) || defined(__arm64__)
-static uint32_t
-get_arm_arch_subinfo(CSSymbolRef symbol)
-{
-	assert(!CSIsNull(symbol));
-	CSSymbolOwnerRef owner = CSSymbolGetSymbolOwner(symbol);
-	CSArchitecture arch = CSSymbolOwnerGetArchitecture(owner);
-
-	if (CSArchitectureMatchesArchitecture(arch, kCSArchitectureArm64_32)) {
-		return FASTTRAP_FN_ARM64_32;
-	}
-	else if (CSArchitectureMatchesArchitecture(arch, kCSArchitectureArm64)) {
-		return FASTTRAP_FN_ARM64;
-	}
-	else if (CSSymbolIsThumb(symbol)) {
-		return FASTTRAP_FN_THUMB;
-	}
-	else if (CSSymbolIsArm(symbol)) {
-		return FASTTRAP_FN_ARM;
-	}
-	return 0;
-}
-#endif
-
 
 /*
  * Search the process symbol tables looking for a symbol whose name matches the
