Seulement dans emacs-70.1.pd1/emacs/src/s: ._darwin.h
diff -ru emacs-70.1/emacs/src/s/darwin.h emacs-70.1.pd1/emacs/src/s/darwin.h
--- emacs-70.1/emacs/src/s/darwin.h	2007-05-15 23:57:33.000000000 +0200
+++ emacs-70.1.pd1/emacs/src/s/darwin.h	2012-09-22 23:34:45.000000000 +0200
@@ -41,16 +41,17 @@
 
 #if 0 /* Don't define DARWIN on Mac OS X because CoreFoundation.h uses
 	 it to distinguish Mac OS X from bare Darwin.  */
-#ifndef	DARWIN
+//#ifndef	DARWIN
 #define	DARWIN 1
-#endif
+//#endif
 #endif
 
 /* MAC_OS is used to conditionally compile code common to both MAC_OS8
    and MAC_OSX.  */
 #ifdef MAC_OSX
 #ifdef HAVE_CARBON
-#define MAC_OS
+//#define MAC_OS
+#undef HAVE_CARBON
 #endif
 #endif
 
diff -ru emacs-70.1/Makefile emacs-70.1.pd1/Makefile
--- emacs-70.1/Makefile	2008-02-09 02:28:25.000000000 +0100
+++ emacs-70.1.pd1/Makefile	2012-09-22 23:58:25.000000000 +0200
@@ -82,8 +82,6 @@
 	$(INSTALL) -s -o root -g wheel -m 4555 $(SYMROOT)/dumpemacs $(DSTROOT)/usr/libexec/dumpemacs
 	$(INSTALL) -s -o root -g wheel -m 555 $(SYMROOT)/emacswrapper $(DSTROOT)/usr/bin/emacs
 #fixup Emacs.app/Contents/Resources/English.lproj/InfoPlist.strings encoding
-	iconv -f US-ASCII -t UTF-16 \
-		$(SRCROOT)/emacs/mac/Emacs.app/Contents/Resources/English.lproj/InfoPlist.strings > $(App_Dir)/Emacs.app/Contents/Resources/English.lproj/InfoPlist.strings
 
 build::
 	@echo "Bootstraping $(Project)..."
@@ -106,7 +104,6 @@
 	$(RM) "$(DSTROOT)/usr/bin/ctags" "$(DSTROOT)/usr/share/man/man1/ctags.1"
 	$(RM) -r "$(DSTROOT)/usr/var"
 	for f in $(EXTRAEL); do $(RM) -f "$(SRCROOT)/$$f"; done
-	$(LN) -sf /usr/bin/emacs "$(App_Dir)/Emacs.app/Contents/MacOS/Emacs"
 
 install-plist:
 	$(INSTALL) -d $(OSV)
Seulement dans emacs-70.1.pd1: ._Makefile
