diff -ru gcc_select-66/Makefile gcc_select-66.pd1/Makefile
--- gcc_select-66/Makefile	2007-05-12 03:16:25.000000000 +0200
+++ gcc_select-66.pd1/Makefile	2012-09-11 03:25:23.695922003 +0200
@@ -18,7 +18,7 @@
 
 PROGS=gcc g++ gcov c++
 
-.PHONY: all install installhdrs installsrc installdoc clean mklinks installsym
+.PHONY: all install installhdrs installsrc installdoc clean mklinks
 
 all: install
 
@@ -36,14 +36,12 @@
 
 install: installhdrs
 
-install: installdoc mklinks $(OBJROOT)/c99 $(OBJROOT)/c89 installsym
+install: installdoc mklinks $(OBJROOT)/c99 $(OBJROOT)/c89
 	mkdir -p $(DSTROOT)/usr/bin
 	install -s -c -m 555 $(OBJROOT)/c99 $(DSTROOT)/usr/bin/c99
 	install -s -c -m 555 $(OBJROOT)/c89 $(DSTROOT)/usr/bin/c89
 	install -c -m 555 $(SRCROOT)/cpp $(DSTROOT)/usr/bin/cpp
 
-installsym: $(OBJROOT)/c99.dSYM $(OBJROOT)/c89.dSYM
-	cp -rp $^ $(SYMROOT)
 
 mklinks:
 	mkdir -p $(DSTROOT)/usr/bin
Seulement dans gcc_select-66.pd1: Makefile~
