diff -ru gnutar-442.0.1/gnutar/src/common.h gnutar-442.0.1.patch/gnutar/src/common.h
--- gnutar-442.0.1/gnutar/src/common.h	2008-09-18 22:50:06.000000000 +0200
+++ gnutar-442.0.1.patch/gnutar/src/common.h	2009-06-10 23:31:24.000000000 +0200
@@ -21,7 +21,7 @@
 #include "tar.h"
 
 #include <TargetConditionals.h>
-#define HAVE_QUARANTINE (!TARGET_OS_EMBEDDED)
+//#define HAVE_QUARANTINE (!TARGET_OS_EMBEDDED)
 
 #if HAVE_QUARANTINE
 #include <quarantine.h>
