Only in iodbc-34.patch/: .Makefile.swp
diff -ru iodbc-34/Makefile iodbc-34.patch/Makefile
--- iodbc-34/Makefile	2007-05-01 02:52:56.000000000 +0200
+++ iodbc-34.patch/Makefile	2009-06-10 23:26:03.000000000 +0200
@@ -26,7 +26,6 @@
 
 Install_Target = install
 Extra_Configure_Flags += --prefix=/usr --with-iodbc-inidir=/Library/ODBC --disable-gui TMPDIR=$(OBJROOT)
-CFLAGS += -Wl,-framework,CoreFoundation
 tesLipo = $(eval "${LIPO} -verify_arch ppc64 $(RC_Install_Prefix)/bin/iodbctest")
 
 fixup:
