diff -ru IOHIDFamily-258.14/IOHIDFamily/IOHIDEventDriver.cpp IOHIDFamily-258.14.pd1/IOHIDFamily/IOHIDEventDriver.cpp
--- IOHIDFamily-258.14/IOHIDFamily/IOHIDEventDriver.cpp	2009-03-05 16:51:59.000000000 -0800
+++ IOHIDFamily-258.14.pd1/IOHIDFamily/IOHIDEventDriver.cpp	2012-09-08 19:31:03.000000000 -0700
@@ -549,8 +549,6 @@
     AbsoluteTime    elementTS;
     AbsoluteTime    reportTS;
     
-    if (!readyForReports())
-        return;
     
     elements = GetElementArray(GetReportHandlerSlot(reportID), reportType);
 
