Only in IOHIDFamily-258.14.pd1/: com.apple.Xcode.0
diff -ru IOHIDFamily-258.14/hidd/hidd.c IOHIDFamily-258.14.pd1/hidd/hidd.c
--- IOHIDFamily-258.14/hidd/hidd.c	2007-07-18 17:04:26.000000000 -0700
+++ IOHIDFamily-258.14.pd1/hidd/hidd.c	2012-09-08 18:57:44.000000000 -0700
@@ -23,6 +23,8 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/hid/IOHIDEventSystem.h>
 
+typedef struct {} *IOHIDEventSystemRef;
+
 int main (int argc, const char * argv[]) 
 {
     IOHIDEventSystemRef eventSystem = IOHIDEventSystemCreate(kCFAllocatorDefault);
