--- IOKitUser-388.2.1/ps.subproj/IOPowerSourcesPrivate.c.origin	2008-11-04 15:44:08.000000000 +0100
+++ IOKitUser-388.2.1/ps.subproj/IOPowerSourcesPrivate.c	2008-11-04 15:44:24.000000000 +0100
@@ -21,6 +21,9 @@
  */
 
 #include <CoreFoundation/CoreFoundation.h>
+#include <CarbonCore/MacTypes.h>
+#include <objc/objc.h>
+
 #include "IOSystemConfiguration.h"
 #include <IOKit/pwr_mgt/IOPMLibPrivate.h>
 #include "IOPowerSources.h"
