--- IOKitUser-388.2.1/pwr_mgt.subproj/IOPMPowerNotifications.c.origin	2008-11-04 15:14:37.000000000 +0100
+++ IOKitUser-388.2.1/pwr_mgt.subproj/IOPMPowerNotifications.c	2008-11-04 15:15:29.000000000 +0100
@@ -27,6 +27,10 @@
  */
  
 #include <CoreFoundation/CoreFoundation.h>
+
+#include <CarbonCore/MacTypes.h>
+#include <objc/objc.h>
+
 #include <SystemConfiguration/SystemConfiguration.h>
 #include <SystemConfiguration/SCPrivate.h>
 #include <IOKit/IOReturn.h>
