--- IOKitUser-388.2.1/graphics.subproj/PEFSupport.c.origin	2008-11-04 14:25:38.000000000 +0100
+++ IOKitUser-388.2.1/graphics.subproj/PEFSupport.c	2008-11-04 14:27:15.000000000 +0100
@@ -28,6 +28,8 @@
     Written by:    Jeffrey Robbin
     Copyright:    © 1994, 1996 by Apple Computer, Inc., all rights reserved.
 */
+#include <CarbonCore/MacTypes.h>
+#include <objc/objc.h>
 
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/IOKitLib.h>
@@ -118,9 +120,9 @@
 * UnpackPiData expands a compressed section into memory.
 *******************************************************************************/
 static OSErr UnpackPiData(
-    LogicalAddress   thePEFPtr,
+    void *   thePEFPtr,
     SectionHeaderPtr sectionHeaderPtr,
-    LogicalAddress * theData)
+    void * * theData)
 {
     int             cntX, cnt, rpt, dcnt, delta;
     unsigned char    op, b;
@@ -312,9 +314,9 @@
 *******************************************************************************/
 static OSStatus GetSymbolFromPEF(
     char *inSymbolName,
-    const LogicalAddress thePEFPtr,
-    LogicalAddress theSymbolPtr,
-    ByteCount theSymbolSize)
+    void * thePEFPtr,
+    void * theSymbolPtr,
+    void * theSymbolSize)
 {
     StringPtr           theSymbolName = (StringPtr) inSymbolName;
     ContainerHeaderPtr  containerHeaderPtr;  // Pointer to the Container Header
@@ -328,7 +330,7 @@
     ExportSymbolEntryPtr       exportSymbolEntryPtr;
     LoaderExportChainEntryPtr  exportChainEntryPtr;
     StringPtr           exportSymbolName;
-    LogicalAddress      expandedDataPtr;
+    void *      expandedDataPtr;
     unsigned char *     sourceDataPtr;
     unsigned char *     destDataPtr;
 
@@ -451,7 +453,7 @@
 /*******************************************************************************
 *
 *******************************************************************************/
-static IOByteCount GetPEFLen(LogicalAddress thePEFPtr)
+static IOByteCount GetPEFLen(void * thePEFPtr)
 {
     ContainerHeaderPtr containerHeaderPtr; // Pointer to the Container Header
     SectionHeaderPtr sections;
@@ -642,7 +644,7 @@
 	    {
 		curDesc = &_curDesc;
 		err = GetSymbolFromPEF(descripName,
-		    (const LogicalAddress)CFDataGetBytePtr(ndrv),
+		    (const void *)CFDataGetBytePtr(ndrv),
 		    curDesc, sizeof(DriverDescription));
 	    }
 	    if (err != noErr)
