diff -ru IOKitUser-388.2.1.origin/IOKitLib.h IOKitUser-388.2.1/IOKitLib.h
--- IOKitUser-388.2.1.origin/IOKitLib.h	2008-10-23 18:36:30.000000000 +0200
+++ IOKitUser-388.2.1/IOKitLib.h	2008-10-23 20:52:26.000000000 +0200
@@ -52,6 +52,10 @@
 
 #include <AvailabilityMacros.h>
 
+#ifndef kNilOptions
+  #define kNilOptions 0
+#endif
+
 __BEGIN_DECLS
 
 /*! @header IOKitLib
diff -ru IOKitUser-388.2.1.origin/graphics.subproj/GetSymbolFromPEF.h IOKitUser-388.2.1/graphics.subproj/GetSymbolFromPEF.h
--- IOKitUser-388.2.1.origin/graphics.subproj/GetSymbolFromPEF.h	2008-10-23 18:36:30.000000000 +0200
+++ IOKitUser-388.2.1/graphics.subproj/GetSymbolFromPEF.h	2008-10-23 20:52:26.000000000 +0200
@@ -60,9 +60,8 @@
     uint32_t    oldImpVersion;    // old implementation version number
     uint32_t    currentVersion;   // current version number
     uint16_t    nbrOfSections;    // nbr of sections (rel. 1)
-    uint16_t    loadableSections; // nbr of loadable sectionsfor execution
-                                       //   (= nbr of 1st non-loadable section)
-    LogicalAddress    memoryAddress;   // location this container was last loaded
+    uint16_t    loadableSections; // nbr of loadable sectionsfor execution //   (= nbr of 1st non-loadable section)
+    //LogicalAddress    memoryAddress;   // location this container was last loaded
 };
 typedef struct ContainerHeader ContainerHeader, *ContainerHeaderPtr;
 
@@ -180,9 +179,9 @@
 
 static OSStatus GetSymbolFromPEF(
     char *theSymbolName,
-    LogicalAddress thePEFPtr,
-    LogicalAddress theSymbolPtr,
-    ByteCount      theSymbolSize);
+    void * thePEFPtr,
+    void * theSymbolPtr,
+    void * theSymbolSize);
 
 static Boolean SymbolCompare(
     StringPtr     theLookedForSymbol,
@@ -190,9 +189,9 @@
     uint32_t  theExportSymbolLength);
                         
 static OSErr UnpackPiData(
-    LogicalAddress   thePEFPtr,
+    void *   thePEFPtr,
     SectionHeaderPtr sectionHeaderPtr,
-    LogicalAddress * theData);
+    void * * theData);
 
 static unsigned char PEFGetNextByte(
     unsigned char ** rawBuffer,
diff -ru IOKitUser-388.2.1.origin/hid.subproj/IOHIDEventSystem.defs IOKitUser-388.2.1/hid.subproj/IOHIDEventSystem.defs
--- IOKitUser-388.2.1.origin/hid.subproj/IOHIDEventSystem.defs	2008-10-23 18:36:30.000000000 +0200
+++ IOKitUser-388.2.1/hid.subproj/IOHIDEventSystem.defs	2008-10-23 20:52:26.000000000 +0200
@@ -0,0 +1,2 @@
+subsystem IOHIDEventSystem 71000;
+
