diff -ru ipv6configuration-27/libraries/service.c ipv6configuration-27.patch/libraries/service.c
--- ipv6configuration-27/libraries/service.c	2007-03-02 01:19:34.000000000 +0100
+++ ipv6configuration-27.patch/libraries/service.c	2009-06-11 00:58:58.000000000 +0200
@@ -798,6 +798,7 @@
  */
 
 #ifndef NO_CFUserNotification
+/*
 static void
 user_confirm(CFUserNotificationRef userNotification, CFOptionFlags responseFlags)
 {
@@ -886,7 +887,7 @@
     my_CFRelease(&url);
     return;
 }
-
+*/
 #else
 
 static void

diff -ru ipv6configuration-27/libraries/service.c ipv6configuration-27.patch/libraries/service.c
--- ipv6configuration-27/libraries/service.c	2009-06-11 00:59:43.000000000 +0200
+++ ipv6configuration-27.patch/libraries/service.c	2009-06-11 01:04:00.000000000 +0200
@@ -805,7 +805,6 @@
     int i;
     int ifstate_count;
 
-    /* clean-up the notification */
     ifstate_count = dynarray_count(&G_ifstate_list);
     for (i = 0; i < ifstate_count; i++) {
         IFState_t *	ifstate = dynarray_element(&G_ifstate_list, i);
