diff -ru KerberosHelper-31.5/Source/KerberosHelper.c KerberosHelper-31.5.pd1/Source/KerberosHelper.c
--- KerberosHelper-31.5/Source/KerberosHelper.c	2009-02-03 04:29:36.000000000 +0100
+++ KerberosHelper-31.5.pd1/Source/KerberosHelper.c	2012-09-13 17:11:46.000000000 +0200
@@ -35,7 +35,8 @@
 
 #include "LKDCHelper.h"
 
-#include <Carbon/Carbon.h>
+//#include <Carbon/Carbon.h>
+#include <CarbonCore/MacErrors.h>
 #include <Security/Security.h>
 #include <Security/SecCertificatePriv.h>
 #include <unistd.h>
diff -ru KerberosHelper-31.5/Source/lookupDSLocalKDC.c KerberosHelper-31.5.pd1/Source/lookupDSLocalKDC.c
--- KerberosHelper-31.5/Source/lookupDSLocalKDC.c	2007-09-12 04:56:28.000000000 +0200
+++ KerberosHelper-31.5.pd1/Source/lookupDSLocalKDC.c	2012-09-13 17:16:44.000000000 +0200
@@ -31,7 +31,8 @@
 
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_5
 
-#include <Carbon/Carbon.h>
+//#include <Carbon/Carbon.h>
+#include <CarbonCore/MacErrors.h>
 #include <OpenDirectory/OpenDirectory.h>
 
 OSStatus DSCopyLocalKDC (CFStringRef *realm) 
diff -ru KerberosHelper-31.5/Source/utils.c KerberosHelper-31.5.pd1/Source/utils.c
--- KerberosHelper-31.5/Source/utils.c	2007-05-08 03:30:32.000000000 +0200
+++ KerberosHelper-31.5.pd1/Source/utils.c	2012-09-13 17:12:33.000000000 +0200
@@ -28,7 +28,8 @@
 
 #include "utils.h"
 
-#include <Carbon/Carbon.h>
+//#include <Carbon/Carbon.h>
+#include <CarbonCore/MacErrors.h>
 
 OSStatus __KRBCreateUTF8StringFromCFString (CFStringRef inString, char **outString)
 {
