--- kext_tools-117/prelink.c.origin	2008-10-30 13:17:02.000000000 +0100
+++ kext_tools-117/prelink.c	2008-10-30 13:19:51.000000000 +0100
@@ -47,6 +47,8 @@
 #include <IOKit/IOCFSerialize.h>
 #include <libkern/OSByteOrder.h>
 
+#include <objc/objc.h>
+#include <CarbonCore/MacTypes.h>
 /*******************************************************************************
 *******************************************************************************/
 
