--- kext_tools-117/kextd_main.c.origin	2008-10-30 13:53:00.000000000 +0100
+++ kext_tools-117/kextd_main.c	2008-10-30 13:54:53.000000000 +0100
@@ -59,6 +59,9 @@
 #include "bootcaches.h"
 #include "utility.h"
 
+#include <objc/objc.h>
+#include <CoreFoundation/CFNotificationCenter.h>
+
 /*******************************************************************************
 * Globals set from invocation arguments (XX could use fewer globals :?).
 *******************************************************************************/
@@ -1209,13 +1212,13 @@
 
    /* Watch for RAID changes so we can forcibly update their boot partitions.
     */
-    CFNotificationCenterAddObserver(CFNotificationCenterGetLocalCenter(),
+    /*CFNotificationCenterAddObserver(CFNotificationCenterGetLocalCenter(),
         NULL, // const void *observer
         updateRAIDSet,
         CFSTR(kAppleRAIDNotificationSetChanged),
         NULL, // const void *object
         CFNotificationSuspensionBehaviorHold);
-    kern_result = AppleRAIDEnableNotifications();
+    kern_result = AppleRAIDEnableNotifications();*/
     if (kern_result != KERN_SUCCESS) {
         kextd_error_log("couldn't register for RAID notifications");
     }
