--- launchd-258.18/launchd/src/launchd.c.origin	2008-10-26 20:16:15.000000000 +0100
+++ launchd-258.18/launchd/src/launchd.c	2008-10-26 20:16:47.000000000 +0100
@@ -288,7 +288,7 @@
 #if HAVE_SECURITY
 	OSStatus (*sescr)(SessionCreationFlags flags, SessionAttributeBits attributes);
 	void *seclib;
-
+	OSStatus noErr=0;
 	if (launchd_assumes((seclib = dlopen(SECURITY_LIB, RTLD_LAZY)) != NULL)) {
 		if (launchd_assumes((sescr = dlsym(seclib, "SessionCreate")) != NULL)) {
 			launchd_assumes(sescr(0, 0) == noErr);
