diff -ru launchd-258.25.pd1/launchd/src/config.h launchd-258.25.pd2/launchd/src/config.h
--- launchd-258.25.pd1/launchd/src/config.h	2008-10-30 02:27:11.000000000 +0100
+++ launchd-258.25.pd2/launchd/src/config.h	2012-09-10 04:28:01.765563608 +0200
@@ -1,7 +1,7 @@
 #ifndef __CONFIG_H__
 #define __CONFIG_H__
 #include <TargetConfig.h>
-#define HAVE_QUARANTINE TARGET_HAVE_QUARANTINE
-#define HAVE_SANDBOX TARGET_HAVE_SANDBOX
-#define HAVE_SECURITY !TARGET_HAVE_EMBEDDED_SECURITY
+#define HAVE_QUARANTINE 0
+#define HAVE_SANDBOX 0
+#define HAVE_SECURITY 0
 #endif /* __CONFIG_H__ */
