diff -ru launchd-258.18.origin/launchd/src/SystemStarter.c launchd-258.18/launchd/src/SystemStarter.c
--- launchd-258.18.origin/launchd/src/SystemStarter.c	2008-10-24 02:49:29.000000000 +0200
+++ launchd-258.18/launchd/src/SystemStarter.c	2008-10-24 03:17:03.000000000 +0200
@@ -151,13 +151,13 @@
 	if (stat("/etc/rc.local", &sb) != -1) {
 		fwexec(_PATH_BSHELL, "/etc/rc.local", NULL);
 	}
-
+/*
 	CFNotificationCenterPostNotificationWithOptions(
 			CFNotificationCenterGetDistributedCenter(),
 			CFSTR("com.apple.startupitems.completed"),
 			NULL, NULL,
 			kCFNotificationDeliverImmediately | kCFNotificationPostToAllSessions);
-
+*/
 	r = kevent(kq, NULL, 0, &kev, 1, NULL);
 	assert(r != -1);
 	assert(kev.filter == EVFILT_SIGNAL && kev.ident == SIGTERM);
diff -ru launchd-258.18.origin/launchd/src/config.h.in launchd-258.18/launchd/src/config.h.in
--- launchd-258.18.origin/launchd/src/config.h.in	2008-10-24 02:49:29.000000000 +0200
+++ launchd-258.18/launchd/src/config.h.in	2008-10-24 03:17:02.000000000 +0200
@@ -90,7 +90,7 @@
 #undef HAVE_PATHS_H
 
 /* Define to 1 if you have the quarantine library */
-#undef HAVE_QUARANTINE
+//#undef HAVE_QUARANTINE
 
 /* Define to 1 if you have the `realloc' function. */
 #undef HAVE_REALLOC
