--- launchd-258.18/launchd/src/launchproxy.c.origin	2008-10-26 20:20:18.000000000 +0100
+++ launchd-258.18/launchd/src/launchproxy.c	2008-10-26 20:20:47.000000000 +0100
@@ -87,6 +87,7 @@
 	launch_data_t tmp, resp, msg = launch_data_alloc(LAUNCH_DATA_STRING);
 	const char *prog = argv[1];
 	bool w = false, dupstdout = true, dupstderr = true;
+	OSStatus noErr = SessionCreate(0, 0);
 
 	launch_data_set_string(msg, LAUNCH_KEY_CHECKIN);
 
