diff -ru launchd-258.1/launchd/src/launchd_core_logic.c launchd-258.1.pd1/launchd/src/launchd_core_logic.c
--- launchd-258.1/launchd/src/launchd_core_logic.c	2007-12-07 22:15:23.000000000 +0100
+++ launchd-258.1.pd1/launchd/src/launchd_core_logic.c	2008-08-27 19:47:55.000000000 +0200
@@ -338,8 +338,6 @@
 	unsigned int lastlookup_gennum;
 	char *seatbelt_profile;
 	uint64_t seatbelt_flags;
-	void *quarantine_data;
-	size_t quarantine_data_sz;
 	pid_t p;
 	int argc;
 	int last_exit_status;
@@ -868,9 +866,6 @@
 	if (j->seatbelt_profile) {
 		free(j->seatbelt_profile);
 	}
-	if (j->quarantine_data) {
-		free(j->quarantine_data);
-	}
 	if (j->j_binpref) {
 		free(j->j_binpref);
 	}
@@ -1554,14 +1549,6 @@
 	switch (key[0]) {
 	case 'q':
 	case 'Q':
-		if (strcasecmp(key, LAUNCH_JOBKEY_QUARANTINEDATA) == 0) {
-			size_t tmpsz = launch_data_get_opaque_size(value);
-
-			if (job_assumes(j, j->quarantine_data = malloc(tmpsz))) {
-				memcpy(j->quarantine_data, launch_data_get_opaque(value), tmpsz);
-				j->quarantine_data_sz = tmpsz;
-			}
-		}
 		break;
 	default:
 		break;
@@ -2680,16 +2667,6 @@
 		job_assumes(j, binpref_out_cnt == j->j_binpref_cnt);
 	}
 
-	if (j->quarantine_data) {
-		qtn_proc_t qp;
-
-		if (job_assumes(j, qp = qtn_proc_alloc())) {
-			if (job_assumes(j, qtn_proc_init_with_data(qp, j->quarantine_data, j->quarantine_data_sz) == 0)) {
-				job_assumes(j, qtn_proc_apply_to_self(qp) == 0);
-			}
-		}
-	}
-
 	if (j->seatbelt_profile) {
 		char *seatbelt_err_buf = NULL;
 
diff -ru launchd-258.1/launchd/src/liblaunch_private.h launchd-258.1.pd1/launchd/src/liblaunch_private.h
--- launchd-258.1/launchd/src/liblaunch_private.h	2007-12-07 22:15:23.000000000 +0100
+++ launchd-258.1.pd1/launchd/src/liblaunch_private.h	2008-08-27 19:48:31.000000000 +0200
@@ -24,7 +24,6 @@
 #include <sys/types.h>
 #include <launch.h>
 #include <unistd.h>
-#include <quarantine.h>
 
 #pragma GCC visibility push(default)
 
@@ -46,7 +45,6 @@
 #define LAUNCH_KEY_BATCHCONTROL		"BatchControl"
 #define LAUNCH_KEY_BATCHQUERY		"BatchQuery"
 
-#define LAUNCH_JOBKEY_QUARANTINEDATA	"QuarantineData"
 #define LAUNCH_JOBKEY_SANDBOXPROFILE	"SandboxProfile"
 #define LAUNCH_JOBKEY_SANDBOXFLAGS	"SandboxFlags"
 #define LAUNCH_JOBKEY_SANDBOX_NAMED	"Named"
@@ -114,7 +112,6 @@
  	mach_port_t *		spawn_observer_port;
  	const cpu_type_t *	spawn_binpref;
 	size_t			spawn_binpref_cnt;
-	qtn_proc_t		spawn_quarantine;
 	const char *		spawn_seatbelt_profile;
 	const uint64_t *	spawn_seatbelt_flags;
 };
diff -ru launchd-258.1/launchd/src/libvproc.c launchd-258.1.pd1/launchd/src/libvproc.c
--- launchd-258.1/launchd/src/libvproc.c	2007-09-14 02:33:43.000000000 +0200
+++ launchd-258.1.pd1/launchd/src/libvproc.c	2008-08-27 19:49:17.000000000 +0200
@@ -201,15 +201,6 @@
 
 	if (spawn_attrs) switch (struct_version) {
 	case 2:
-		if (spawn_attrs->spawn_quarantine) {
-			char qbuf[QTN_SERIALIZED_DATA_MAX];
-			size_t qbuf_sz = QTN_SERIALIZED_DATA_MAX;
-
-			if (qtn_proc_to_data(spawn_attrs->spawn_quarantine, qbuf, &qbuf_sz) == 0) {
-				tmp = launch_data_new_opaque(qbuf, qbuf_sz);
-				launch_data_dict_insert(in_obj, tmp, LAUNCH_JOBKEY_QUARANTINEDATA);
-			}
-		}
 
 		if (spawn_attrs->spawn_seatbelt_profile) {
 			tmp = launch_data_new_string(spawn_attrs->spawn_seatbelt_profile);
