diff -ru ld64-77.1/src/ld.cpp ld64-77.1.pd1/src/ld.cpp
--- ld64-77.1/src/ld.cpp	2008-02-12 21:38:12.000000000 +0100
+++ ld64-77.1.pd1/src/ld.cpp	2012-09-10 02:09:19.221728929 +0200
@@ -1403,6 +1403,7 @@
 
 void Linker::processDTrace()
 {
+	return;
 	// handle dtrace 2.0 static probes
 	if ( (fOptions.outputKind() != Options::kObjectFile) && ((fDtraceProbeSites.size() != 0) || (fDtraceIsEnabledSites.size() != 0)) ) {
 		// partition probes by provider name
diff -ru ld64-77.1/src/MachOWriterExecutable.hpp ld64-77.1.pd1/src/MachOWriterExecutable.hpp
--- ld64-77.1/src/MachOWriterExecutable.hpp	2007-10-27 02:09:05.000000000 +0200
+++ ld64-77.1.pd1/src/MachOWriterExecutable.hpp	2012-09-10 02:07:51.113730679 +0200
@@ -36,7 +36,7 @@
 #include <mach-o/stab.h>
 #include <uuid/uuid.h>
 #include <mach/i386/thread_status.h>
-#include <mach/ppc/thread_status.h>
+//#include <mach/ppc/thread_status.h>
 #include <CommonCrypto/CommonDigest.h>
 
 #include <vector>
