diff -ru Libc-498.1.7/Makefile.features Libc-498.1.7.pd1/Makefile.features
--- Libc-498.1.7/Makefile.features	2008-09-30 02:35:24.000000000 +0200
+++ Libc-498.1.7.pd1/Makefile.features	2012-09-10 00:42:19.885832606 +0200
@@ -78,7 +78,7 @@
 	@echo '/* #undef PR_5243343 */' >> ${.TARGET}
 .endif
 .ifdef FEATURE_PLOCKSTAT
-	@echo '#define PLOCKSTAT 1' >> ${.TARGET}
+	@echo '/* #undef PLOCKSTAT */' >> ${.TARGET}
 .else
 	@echo '/* #undef PLOCKSTAT */' >> ${.TARGET}
 .endif
Seulement dans Libc-498.1.7.pd1: Makefile.features~
diff -ru Libc-498.1.7/pthreads/Makefile.inc Libc-498.1.7.pd1/pthreads/Makefile.inc
--- Libc-498.1.7/pthreads/Makefile.inc	2008-07-16 00:35:58.000000000 +0200
+++ Libc-498.1.7.pd1/pthreads/Makefile.inc	2012-09-10 00:38:55.737836662 +0200
@@ -7,15 +7,6 @@
 MISRCS += pthread_cancelable.c pthread_cond.c pthread_tsd.c pthread.c \
         pthread_mutex.c    thread_setup.c   stack.s pthread_rwlock.c
 
-.ifdef FEATURE_PLOCKSTAT
-${SYMROOTINC}/plockstat.h: ${.CURDIR}/pthreads/plockstat.d
-	mkdir -p ${SYMROOTINC}
-	dtrace -o ${.TARGET} -C -h -s ${.ALLSRC}
-
-.for _src in pthread_cond.c pthread_mutex.c pthread_rwlock.c
-${_src:R}.${OBJSUFFIX}: ${SYMROOTINC}/plockstat.h
-.endfor
-.endif # FEATURE_PLOCKSTAT
 
 PTHREADS_INSTHDRS += pthread.h pthread_impl.h sched.h
 PTHREADS_INSTHDRS := ${PTHREADS_INSTHDRS:S/^/${.CURDIR}\/pthreads\//}
Seulement dans Libc-498.1.7.pd1/pthreads: Makefile.inc~
diff -ru Libc-498.1.7/pthreads/pthread_mutex.c Libc-498.1.7.pd1/pthreads/pthread_mutex.c
--- Libc-498.1.7/pthreads/pthread_mutex.c	2008-07-16 00:35:58.000000000 +0200
+++ Libc-498.1.7.pd1/pthreads/pthread_mutex.c	2012-09-10 00:34:53.937841465 +0200
@@ -52,7 +52,8 @@
 
 #include "pthread_internals.h"
 
-#ifdef PLOCKSTAT
+#if 0
+//#ifdef PLOCKSTAT
 #include "plockstat.h"
 #else /* !PLOCKSTAT */
 #define	PLOCKSTAT_MUTEX_SPIN(x)
