commit 860148384dbee82df190694ac7a0103f6d3ba704
Author: William Kent <wjk@users.noreply.github.com>
Date:   Tue Dec 11 15:04:28 2018 -0500

    Apply patch

diff --git a/xcodeconfig/BSD.xcconfig b/xcodeconfig/BSD.xcconfig
new file mode 100644
index 0000000..4348218
--- /dev/null
+++ b/xcodeconfig/BSD.xcconfig
@@ -0,0 +1,11 @@
+ARCHS = $(ARCHS_STANDARD_32_64_BIT);
+CODE_SIGN_IDENTITY = -;
+CURRENT_PROJECT_VERSION = $(RC_ProjectSourceVersion);
+DEAD_CODE_STRIPPING = YES;
+DEBUG_INFORMATION_FORMAT = dwarf-with-dsym;
+PREBINDING = NO;
+// Current Mac OS
+SDKROOT = macosx;
+//USE_HEADERMAP = NO;
+VERSION_INFO_PREFIX = __attribute__((visibility("hidden"))) __
+VERSIONING_SYSTEM = apple-generic;
diff --git a/xcodeconfig/libdispatch.xcconfig b/xcodeconfig/libdispatch.xcconfig
index 2c825f7..a06ee26 100644
--- a/xcodeconfig/libdispatch.xcconfig
+++ b/xcodeconfig/libdispatch.xcconfig
@@ -18,10 +18,8 @@
 // @APPLE_APACHE_LICENSE_HEADER_END@
 //
 
-#include "<DEVELOPER_DIR>/Makefiles/CoreOS/Xcode/BSD.xcconfig"
-#include "<DEVELOPER_DIR>/AppleInternal/XcodeConfig/PlatformSupport.xcconfig"
+#include "BSD.xcconfig"
 
-SDKROOT = macosx.internal
 SUPPORTED_PLATFORMS = macosx iphoneos iphonesimulator appletvos appletvsimulator watchos watchsimulator
 PRODUCT_NAME = libdispatch
 EXECUTABLE_PREFIX =
diff --git a/xcodeconfig/libfirehose.xcconfig b/xcodeconfig/libfirehose.xcconfig
index 4c71199..418045d 100644
--- a/xcodeconfig/libfirehose.xcconfig
+++ b/xcodeconfig/libfirehose.xcconfig
@@ -18,7 +18,7 @@
 // @APPLE_APACHE_LICENSE_HEADER_END@
 //
 
-SUPPORTED_PLATFORMS = macosx iphoneos iphonesimulator appletvos appletvsimulator watchos watchsimulator
+SUPPORTED_PLATFORMS = macosx
 PRODUCT_NAME = $(TARGET_NAME)
 INSTALL_PATH = /usr/local/lib/
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) FIREHOSE_SERVER=1 DISPATCH_USE_DTRACE=0
diff --git a/xcodeconfig/libfirehose_kernel.xcconfig b/xcodeconfig/libfirehose_kernel.xcconfig
index c572f80..e69108f 100644
--- a/xcodeconfig/libfirehose_kernel.xcconfig
+++ b/xcodeconfig/libfirehose_kernel.xcconfig
@@ -20,14 +20,14 @@
 
 #include "libfirehose.xcconfig"
 
-SUPPORTED_PLATFORMS = macosx iphoneos appletvos watchos
+SUPPORTED_PLATFORMS = macosx
 PRODUCT_NAME = $(TARGET_NAME)
 INSTALL_PATH = /usr/local/lib/kernel/
 GCC_PREPROCESSOR_DEFINITIONS = $(inherited) KERNEL=1 DISPATCH_USE_DTRACE=0
 OTHER_CFLAGS = -mkernel -nostdinc -Wno-packed
 // LLVM_LTO = YES
 PRIVATE_HEADERS_FOLDER_PATH = /usr/local/include/kernel/os
-HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(SDKROOT)/System/Library/Frameworks/Kernel.framework/PrivateHeaders $(SDKROOT)/System/Library/Frameworks/Kernel.framework/Headers $(SDKROOT)/usr/local/include/os $(SDKROOT)/usr/local/include/firehose
+HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(SDKROOT)/System/Library/Frameworks/Kernel.framework/PrivateHeaders $(DEPENDENCIES_DIR)/System/Library/Frameworks/Kernel.framework/PrivateHeaders $(DEPENDENCIES_DIR)/usr/local/include/os $(DEPENDENCIES_DIR)/usr/local/include/firehose $(SDKROOT)/usr/include
 
 COPY_HEADERS_RUN_UNIFDEF = YES
 COPY_HEADERS_UNIFDEF_FLAGS = -DKERNEL=1 -DOS_FIREHOSE_SPI=1 -DOS_VOUCHER_ACTIVITY_SPI_TYPES=1 -UOS_VOUCHER_ACTIVITY_SPI
