diff -ru libgcc-8.1/libgcc/configure libgcc-8.1.pd1/libgcc/configure
--- libgcc-8.1/libgcc/configure	2007-05-14 05:19:11.000000000 +0200
+++ libgcc-8.1.pd1/libgcc/configure	2012-09-09 23:41:42.089904868 +0200
@@ -3349,9 +3349,9 @@
 fi
 
 # Default to --enable-multilib.
-if test x${enable_multilib} = x ; then
-  target_configargs="--enable-multilib ${target_configargs}"
-fi
+#if test x${enable_multilib} = x ; then
+  target_configargs="--disable-multilib ${target_configargs}"
+#fi
 
 # Pass --with-newlib if appropriate.  Note that target_configdirs has
 # changed from the earlier setting of with_newlib.
Seulement dans libgcc-8.1.pd1/libgcc: configure~
diff -ru libgcc-8.1/libgcc/configure.in libgcc-8.1.pd1/libgcc/configure.in
--- libgcc-8.1/libgcc/configure.in	2006-12-29 18:47:06.000000000 +0100
+++ libgcc-8.1.pd1/libgcc/configure.in	2012-09-09 23:41:59.621904520 +0200
@@ -2056,9 +2056,9 @@
 fi
 
 # Default to --enable-multilib.
-if test x${enable_multilib} = x ; then
-  target_configargs="--enable-multilib ${target_configargs}"
-fi
+#if test x${enable_multilib} = x ; then
+  target_configargs="--disable-multilib ${target_configargs}"
+#fi
 
 # Pass --with-newlib if appropriate.  Note that target_configdirs has
 # changed from the earlier setting of with_newlib.
Seulement dans libgcc-8.1.pd1/libgcc: configure.in~
diff -ru libgcc-8.1/libgcc/gcc/configure libgcc-8.1.pd1/libgcc/gcc/configure
--- libgcc-8.1/libgcc/gcc/configure	2007-01-02 04:44:31.000000000 +0100
+++ libgcc-8.1.pd1/libgcc/gcc/configure	2012-09-09 23:43:48.529902355 +0200
@@ -7013,12 +7013,12 @@
 
 # Determine whether or not multilibs are enabled.
 # Check whether --enable-multilib or --disable-multilib was given.
-if test "${enable_multilib+set}" = set; then
-  enableval="$enable_multilib"
+#if test "${enable_multilib+set}" = set; then
+#  enableval="$enable_multilib"
 
-else
-  enable_multilib=yes
-fi;
+#else
+  enable_multilib=no
+#fi;
 
 
 # Enable __cxa_atexit for C++.
Seulement dans libgcc-8.1.pd1/libgcc/gcc: configure~
diff -ru libgcc-8.1/libgcc/gcc/configure.ac libgcc-8.1.pd1/libgcc/gcc/configure.ac
--- libgcc-8.1/libgcc/gcc/configure.ac	2007-01-02 04:44:31.000000000 +0100
+++ libgcc-8.1.pd1/libgcc/gcc/configure.ac	2012-09-09 23:45:03.373900870 +0200
@@ -671,10 +671,10 @@
 stabs=no)
 
 # Determine whether or not multilibs are enabled.
-AC_ARG_ENABLE(multilib,
-[  --enable-multilib       enable library support for multiple ABIs],
-[], [enable_multilib=yes])
-AC_SUBST(enable_multilib)
+#AC_ARG_ENABLE(multilib,
+#[  --enable-multilib       enable library support for multiple ABIs],
+#[], [enable_multilib=no])
+# AC_SUBST(enable_multilib)
 
 # Enable __cxa_atexit for C++.
 AC_ARG_ENABLE(__cxa_atexit,
Seulement dans libgcc-8.1.pd1/libgcc/gcc: configure.ac~
