diff -ru libgcc-8.1.pd1/Makefile libgcc-8.1.pd2/Makefile
--- libgcc-8.1.pd1/Makefile	2007-10-05 23:14:36.000000000 +0200
+++ libgcc-8.1.pd2/Makefile	2012-09-09 23:53:12.881891146 +0200
@@ -71,7 +71,6 @@
 post-install:
 	$(RM) -r $(DSTROOT)/usr/{bin,include,libexec,share,lib/gcc}
 	dsymutil $(DSTROOT)/usr/lib/libgcc_s.1.dylib
-	mv $(DSTROOT)/usr/lib/libgcc_s.1.dylib.dSYM $(SYMROOT)/
 	strip -x $(DSTROOT)/usr/lib/libgcc_s.1.dylib
 	ln -s libgcc_s.1.dylib  $(DSTROOT)/usr/lib/libgcc_s.1.0.dylib
 
Seulement dans libgcc-8.1.pd2/: Makefile~
