diff -ru Libsystem-111.1.4/Makefile Libsystem-111.1.4.pd1/Makefile
--- Libsystem-111.1.4/Makefile	2008-09-30 04:47:45.000000000 +0200
+++ Libsystem-111.1.4.pd1/Makefile	2012-09-10 01:28:52.813777126 +0200
@@ -21,8 +21,8 @@
 LIBS = -lc -lcommonCrypto -ldyldapis\
        -linfo -ldns_sd -lm -lmacho\
        -lnotify -lkeymgr -llaunch \
-       -lcopyfile -lsandbox -lremovefile
-CONDITIONALLIBS = unc quarantine
+       -lcopyfile -lremovefile
+CONDITIONALLIBS = unc
 .for L in $(CONDITIONALLIBS)
 # the following should be replaced with tconf when 5784037 is complete
 .if exists($(LIBSYS)/lib$(L).a)
Seulement dans Libsystem-111.1.4.pd1: Makefile~
