diff -ru libsecurity_authorization-32564/lib/Authorization.cpp libsecurity_authorization-32564.pd1/lib/Authorization.cpp
--- libsecurity_authorization-32564/lib/Authorization.cpp	2007-08-29 22:26:41.000000000 +0200
+++ libsecurity_authorization-32564.pd1/lib/Authorization.cpp	2012-09-12 00:18:56.682427966 +0200
@@ -42,7 +42,9 @@
 #include <security_cdsa_utilities/AuthorizationWalkers.h>
 #include <securityd_client/ssclient.h>
 #include <CoreFoundation/CFPreferences.h>
-#include <Carbon/../Frameworks/HIToolbox.framework/Headers/TextInputSources.h>
+//#include <Carbon/../Frameworks/HIToolbox.framework/Headers/TextInputSources.h>
+#include <CoreFoundation/CFNotificationCenter.h>
+
 
 #include <security_utilities/logging.h>
 
