diff -ru mDNSResponder-176.3/mDNSMacOSX/helper.c mDNSResponder-176.3.pd1/mDNSMacOSX/helper.c
--- mDNSResponder-176.3/mDNSMacOSX/helper.c	2008-01-30 20:01:51.000000000 +0100
+++ mDNSResponder-176.3.pd1/mDNSMacOSX/helper.c	2012-09-23 01:23:13.372244527 +0200
@@ -126,10 +126,10 @@
 #include "helper-server.h"
 #include "ipsec_options.h"
 
-#if TARGET_OS_EMBEDDED
+//#if TARGET_OS_EMBEDDED
 #define NO_CFUSERNOTIFICATION 1
 #define NO_SECURITYFRAMEWORK 1
-#endif
+//#endif
 
 typedef struct sadb_x_policy *ipsec_policy_t;
 
diff -ru mDNSResponder-176.3/mDNSMacOSX/helper-main.c mDNSResponder-176.3.pd1/mDNSMacOSX/helper-main.c
--- mDNSResponder-176.3/mDNSMacOSX/helper-main.c	2008-03-13 21:55:16.000000000 +0100
+++ mDNSResponder-176.3.pd1/mDNSMacOSX/helper-main.c	2012-09-23 01:23:11.124244572 +0200
@@ -90,12 +90,12 @@
 #include "helpermsg.h"
 #include "helpermsgServer.h"
 
-#if TARGET_OS_EMBEDDED
+//#if TARGET_OS_EMBEDDED
 #include <bootstrap_priv.h>
 #define NO_SECURITYFRAMEWORK 1
 
 #define bootstrap_register(A,B,C) bootstrap_register2((A),(B),(C),0)
-#endif
+//#endif
 
 #ifndef LAUNCH_JOBKEY_MACHSERVICES
 #define LAUNCH_JOBKEY_MACHSERVICES "MachServices"
diff -ru mDNSResponder-176.3/mDNSMacOSX/mDNSMacOSX.c mDNSResponder-176.3.pd1/mDNSMacOSX/mDNSMacOSX.c
--- mDNSResponder-176.3/mDNSMacOSX/mDNSMacOSX.c	2008-07-30 03:08:17.000000000 +0200
+++ mDNSResponder-176.3.pd1/mDNSMacOSX/mDNSMacOSX.c	2012-09-23 01:23:08.412244625 +0200
@@ -806,10 +806,10 @@
 #include <netinet6/in6_var.h>       // For IN6_IFF_NOTREADY etc.
 #include <netinet6/nd6.h>           // For ND6_INFINITE_LIFETIME etc.
 
-#if TARGET_OS_EMBEDDED
+//#if TARGET_OS_EMBEDDED
 #define NO_SECURITYFRAMEWORK 1
 #define NO_CFUSERNOTIFICATION 1
-#endif
+//#endif
 
 #ifndef NO_SECURITYFRAMEWORK
 #include <Security/SecureTransport.h>
