diff -ru mDNSResponder-176.3.pd1/mDNSMacOSX/helper.c mDNSResponder-176.3.pd2/mDNSMacOSX/helper.c
--- mDNSResponder-176.3.pd1/mDNSMacOSX/helper.c	2008-01-30 20:01:51.000000000 +0100
+++ mDNSResponder-176.3.pd2/mDNSMacOSX/helper.c	2012-09-13 21:05:19.235226222 +0200
@@ -92,6 +92,8 @@
 <rdar://problem/5197869> Security: Run mDNSResponder as user id mdnsresponder instead of root
  */
 
+#include <CarbonCore/MacTypes.h>
+#include <CoreFoundation/CFUserNotification.h>
 #include <sys/cdefs.h>
 #include <arpa/inet.h>
 #include <bsm/libbsm.h>
