--- mDNSResponder-176.2/mDNSMacOSX/mDNSMacOSX.c.origin	2008-11-08 00:51:44.000000000 +0100
+++ mDNSResponder-176.2/mDNSMacOSX/mDNSMacOSX.c	2008-11-08 00:53:57.000000000 +0100
@@ -773,6 +773,9 @@
 
 #define USE_V6_ONLY_WHEN_NO_ROUTABLE_V4 0
 
+
+#include <CarbonCore/MacTypes.h>
+#include <CoreFoundation/CFUserNotification.h>
 #include "mDNSEmbeddedAPI.h"          // Defines the interface provided to the client layer above
 #include "DNSCommon.h"
 #include "uDNS.h"
