--- mDNSResponder-176.2/mDNSMacOSX/../mDNSShared/uds_daemon.c.origin	2008-11-08 11:52:11.000000000 +0100
+++ mDNSResponder-176.2/mDNSMacOSX/../mDNSShared/uds_daemon.c	2008-11-08 11:52:32.000000000 +0100
@@ -3955,9 +3955,9 @@
 	// other overly-large structures instead of having a pointer to them, can inadvertently
 	// cause structure sizes (and therefore memory usage) to balloon unreasonably.
 	char sizecheck_request_state          [(sizeof(request_state)           <= 1800) ? 1 : -1];
-	char sizecheck_registered_record_entry[(sizeof(registered_record_entry) <=   30) ? 1 : -1];
-	char sizecheck_service_instance       [(sizeof(service_instance)        <= 6000) ? 1 : -1];
+	//char sizecheck_registered_record_entry[(sizeof(registered_record_entry) <=   30) ? 1 : -1];
+	//char sizecheck_service_instance       [(sizeof(service_instance)        <= 6000) ? 1 : -1];
 	char sizecheck_browser_t              [(sizeof(browser_t)               <= 1000) ? 1 : -1];
 	char sizecheck_reply_hdr              [(sizeof(reply_hdr)               <=   20) ? 1 : -1];
-	char sizecheck_reply_state            [(sizeof(reply_state)             <=   40) ? 1 : -1];
+	//char sizecheck_reply_state            [(sizeof(reply_state)             <=   40) ? 1 : -1];
 	};
