diff -ru network_cmds-307.1.1/Makefile network_cmds-307.1.1.pd1/Makefile
--- network_cmds-307.1.1/Makefile	2008-09-12 01:49:43.000000000 +0200
+++ network_cmds-307.1.1.pd1/Makefile	2012-09-10 02:16:41.513720142 +0200
@@ -1,8 +1,5 @@
 Project = network_cmds
 
-Embedded=$(shell tconf --test TARGET_OS_EMBEDDED)
-
-ifeq ($(Embedded),YES)
 SubProjects = arp.tproj \
 	alias \
 	ifconfig.tproj netstat.tproj\
@@ -10,18 +7,6 @@
 	route.tproj\
 	traceroute.tproj\
         natd.tproj
-else
-SubProjects = arp.tproj \
-	alias \
-	bootparams \
-        ifconfig.tproj netstat.tproj\
-        ping.tproj rarpd.tproj\
-        route.tproj routed.tproj\
-        slattach.tproj spray.tproj\
-        traceroute.tproj trpt.tproj\
-        natd.tproj ipfw.tproj\
-        ping6.tproj traceroute6.tproj rtsol.tproj ndp.tproj rtadvd.tproj\
-	ip6conf.tproj ip6fw.tproj kdumpd.tproj
-endif
+
 
 include $(MAKEFILEPATH)/CoreOS/ReleaseControl/BSDCommon.make
Seulement dans network_cmds-307.1.1.pd1: Makefile~
