diff -ru OpenDirectory-39/CFOpenDirectory.c OpenDirectory-39.pd1/CFOpenDirectory.c
--- OpenDirectory-39/CFOpenDirectory.c	2007-09-15 04:14:58.000000000 +0200
+++ OpenDirectory-39.pd1/CFOpenDirectory.c	2012-09-12 16:39:24.000000000 +0200
@@ -36,12 +36,18 @@
 #include <objc/objc-runtime.h>
 #include <membership.h>
 #include <membershipPriv.h>
-#include <CoreFoundation/CFBridgingPriv.h>
+#include <CoreFoundation/CoreFoundation.h>
 #include <CoreFoundation/CFError.h>
 #include <pwd.h>
 #include <Kerberos/Kerberos.h>
 #include <DirectoryServiceCore/CSharedData.h>      // for custom call
 
+//http://darwinbuild.macosforge.org/trac/ticket/50
+#define CF_IS_OBJC(x,...) FALSE
+#define CF_OBJC_CALL(x,...) do { } while (0)
+#define CF_OBJC_VOIDCALL(x,...) do { } while (0)
+#define CF_OBJC_FUNCDISPATCH(x,...) do { } while (0)
+
 #ifndef kDSConfigRecordsType
     #define kDSConfigRecordsType        "dsConfigType::RecordTypes"
 #endif
