diff -ru pam_modules-36.2/pam_serialnumber/GNUmakefile pam_modules-36.2.pd1/pam_serialnumber/GNUmakefile
--- pam_modules-36.2/pam_serialnumber/GNUmakefile	2007-02-20 21:18:32.000000000 +0100
+++ pam_modules-36.2.pd1/pam_serialnumber/GNUmakefile	2012-09-12 16:01:47.857304232 +0200
@@ -3,6 +3,6 @@
 ##
 
 PAMModuleName            = pam_serialnumber
-Extra_LD_Flags           = -L/usr/local/lib -lAppleSystemInfo -framework CoreFoundation -framework IOKit
+Extra_LD_Flags           = -L/usr/local/lib -framework CoreFoundation -framework IOKit
 
 include ../PAMModule.defs
Seulement dans pam_modules-36.2.pd1/pam_serialnumber: GNUmakefile~
diff -ru pam_modules-36.2/pam_serialnumber/Makefile pam_modules-36.2.pd1/pam_serialnumber/Makefile
--- pam_modules-36.2/pam_serialnumber/Makefile	2007-02-20 21:18:32.000000000 +0100
+++ pam_modules-36.2.pd1/pam_serialnumber/Makefile	2012-09-12 15:58:29.545308171 +0200
@@ -27,6 +27,6 @@
 SRCS=		pam_serialnumber.c
 MAN=		pam_serialnumber.8
 
-LDADD+=		-L/usr/local/lib -lAppleSystemInfo -framework CoreFoundation -framework IOKit
+LDADD+=		-L/usr/local/lib -framework CoreFoundation -framework IOKit
 
 .include <bsd.lib.mk>
Seulement dans pam_modules-36.2.pd1/pam_serialnumber: Makefile~
diff -ru pam_modules-36.2/pam_serialnumber/pam_serialnumber.c pam_modules-36.2.pd1/pam_serialnumber/pam_serialnumber.c
--- pam_modules-36.2/pam_serialnumber/pam_serialnumber.c	2007-11-29 23:50:37.000000000 +0100
+++ pam_modules-36.2.pd1/pam_serialnumber/pam_serialnumber.c	2012-09-12 15:58:13.369308491 +0200
@@ -37,7 +37,7 @@
 #include <pam/_pam_macros.h>
 #include <pam/pam_mod_misc.h>
 #include <CoreFoundation/CoreFoundation.h>
-#include <AppleSystemInfo/ASI_SerialNumber.h>
+//#include <AppleSystemInfo/ASI_SerialNumber.h>
 #include <sys/mount.h>
 #include <syslog.h>
 
@@ -87,8 +87,9 @@
 		return PAM_AUTH_ERR;
 	}
 	
+	/* For puredarwin : Simply bypass this private lib */
 	/* Get the first 8 characters of the serial number */
-	cfs = ASI_CopyFormattedSerialNumber();
+	cfs = NULL;//ASI_CopyFormattedSerialNumber();
 	if (NULL != cfs && 0 < CFStringGetLength(cfs)) {
 		snbuffsz = CFStringGetLength(cfs)+1;
 		if ((2 > snbuffsz) && (128 < snbuffsz)) {
