diff -ru perl-51.1.4/fix/darwin.sh.ed perl-51.1.4.pd1/fix/darwin.sh.ed
--- perl-51.1.4/fix/darwin.sh.ed	2007-06-19 21:34:10.000000000 +0200
+++ perl-51.1.4.pd1/fix/darwin.sh.ed	2012-09-22 05:38:41.965165504 +0200
@@ -8,7 +8,7 @@
        *-mmacosx-version-min=*) ;;
        *)
            productVersion=`sw_vers -productVersion`
-           ld="${ld} -mmacosx-version-min=${productVersion}"
+           ld="${ld} -mmacosx-version-min=10.5.8"
 	   ;;
    esac
    ;;
Seulement dans perl-51.1.4.pd1/fix: darwin.sh.ed~
diff -ru perl-51.1.4/GNUmakefile perl-51.1.4.pd1/GNUmakefile
--- perl-51.1.4/GNUmakefile	2009-03-11 21:38:58.000000000 +0100
+++ perl-51.1.4.pd1/GNUmakefile	2012-09-22 05:45:31.789157361 +0200
@@ -133,7 +133,6 @@
 	chmod u+w $(PROJECT)/hints/darwin.sh $(PROJECT)/handy.h \
 	    $(PROJECT)/lib/Pod/Perldoc.pm $(PROJECT)/perl.c && \
 	cat hints.append >> $(PROJECT)/hints/darwin.sh && \
-	ed - $(PROJECT)/cop.h < fix/cop.h.ed && \
 	ed - $(PROJECT)/handy.h < fix/handy.h.ed && \
 	ed - $(PROJECT)/hints/darwin.sh < fix/darwin.sh.ed && \
 	ed - $(PROJECT)/lib/Pod/Perldoc.pm < fix/lib_Pod_Perldoc.pm.ed && \
@@ -141,7 +140,6 @@
 	ed - $(PROJECT)/perl.c < fix/perl.c.ed && \
 	sed 's/@VERSION@/$(VERSION)' fix/README.macosx.ed | ed - $(PROJECT)/README.macosx && \
 	ed - $(PROJECT)/util.c < fix/util.c.ed
-	dtrace -h -s $(SRCROOT)/$(FIX)/pldtrace.d -o '$(OBJROOT)/$(PLDTRACE_H)'
 	cd '$(OBJROOT)/$(PROJECT)' && patch -p0 -i $(SRCROOT)/$(FIX)/CVE-2007-5116-regcomp.diff
 	cd '$(OBJROOT)/$(PROJECT)' && patch -p0 -i $(SRCROOT)/$(FIX)/CVE-2008-1927-double-free_5.8.8.patch
 
