Seulement dans PowerManagement-158.10.9.pd1/AppleSmartBatteryManager: ._AppleSmartBattery.cpp
diff -ru PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBattery.cpp PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBattery.cpp
--- PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBattery.cpp	2009-08-07 22:00:52.000000000 +0200
+++ PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBattery.cpp	2012-09-22 11:10:13.000000000 +0200
@@ -597,7 +597,7 @@
     uint32_t    delay_for = 0;
     IOSMBusStatus transaction_status = kIOSMBusStatusPECError;
     bool        transaction_needs_retry = false;
-    char        recv_str[kIOSMBusMaxDataCount+1];
+    char        recv_str[kIOSMBusMaxDataCount+1];//32 ?
     OSNumber    *cell_volt_num = NULL;
     OSNumber    *pfstatus_num = NULL;
 
diff -ru PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBattery.h PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBattery.h
--- PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBattery.h	2008-10-03 02:58:59.000000000 +0200
+++ PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBattery.h	2012-09-22 11:17:00.549882370 +0200
@@ -26,7 +26,8 @@
 
 #include <IOKit/IOService.h>
 #include <IOKit/pwr_mgt/IOPMPowerSource.h>
-#include <IOKit/smbus/IOSMBusController.h>
+//#include <IOKit/smbus/IOSMBusController.h>
+#include "IOSMBusController.h"
 #include <IOKit/acpi/IOACPIPlatformDevice.h>
 
 #include "AppleSmartBatteryCommands.h"
Seulement dans PowerManagement-158.10.9.pd1/AppleSmartBatteryManager: ._AppleSmartBatteryManager.cpp
diff -ru PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBatteryManager.cpp PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBatteryManager.cpp
--- PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBatteryManager.cpp	2008-06-12 02:00:37.000000000 +0200
+++ PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBatteryManager.cpp	2012-09-22 10:57:46.000000000 +0200
@@ -26,6 +26,8 @@
 #include "AppleSmartBatteryManager.h"
 #include "AppleSmartBattery.h"
 
+#include <IOKit/IOCommandPool.h>
+
 #define kMaxRetries     5
 
 static uint8_t  retryDelaysTable[kMaxRetries] =
@@ -384,7 +386,7 @@
        poll of battery state.   */
 
     if(!alarm) return kIOReturnSuccess;
-    
+#if 0    
     if( (kIOMessageSMBusAlarm == type) 
         && (kSMBusManagerAddr == alarm->fromAddress)
         && fBatteryGate)
@@ -413,7 +415,7 @@
                                NULL, NULL, NULL, NULL);
         }
     }
-
+#endif
     return kIOReturnSuccess;
 }
 
diff -ru PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBatteryManager.h PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBatteryManager.h
--- PowerManagement-158.10.9/AppleSmartBatteryManager/AppleSmartBatteryManager.h	2008-06-12 02:00:37.000000000 +0200
+++ PowerManagement-158.10.9.pd1/AppleSmartBatteryManager/AppleSmartBatteryManager.h	2012-09-22 11:16:57.621882428 +0200
@@ -25,7 +25,8 @@
 #define __AppleSmartBatteryManager__
 
 #include <IOKit/IOService.h>
-#include <IOKit/smbus/IOSMBusController.h>
+//#include <IOKit/smbus/IOSMBusController.h>
+#include "IOSMBusController.h"
 #include "AppleSmartBattery.h"
 #include "AppleSmartBatteryManagerUserClient.h"
 
Seulement dans PowerManagement-158.10.9.pd1/AppleSmartBatteryManager: IOSMBusController.h
Seulement dans PowerManagement-158.10.9.pd1: build
Seulement dans PowerManagement-158.10.9.pd1: com.apple.Xcode.0
Seulement dans PowerManagement-158.10.9.pd1/pmconfigd: ._PrivateLib.h
diff -ru PowerManagement-158.10.9/pmconfigd/PrivateLib.h PowerManagement-158.10.9.pd1/pmconfigd/PrivateLib.h
--- PowerManagement-158.10.9/pmconfigd/PrivateLib.h	2009-08-07 22:00:52.000000000 +0200
+++ PowerManagement-158.10.9.pd1/pmconfigd/PrivateLib.h	2012-09-22 11:02:23.000000000 +0200
@@ -24,6 +24,7 @@
 #define _privatelib_h_
 
 #include <TargetConditionals.h>
+#include <CoreFoundation/CFUserNotification.h>
 
 #if !TARGET_OS_EMBEDDED
   #define HAVE_CF_USER_NOTIFICATION     1
diff -ru PowerManagement-158.10.9/pmconfigd/TTYKeepAwake.c PowerManagement-158.10.9.pd1/pmconfigd/TTYKeepAwake.c
--- PowerManagement-158.10.9/pmconfigd/TTYKeepAwake.c	2008-02-01 04:28:46.000000000 +0100
+++ PowerManagement-158.10.9.pd1/pmconfigd/TTYKeepAwake.c	2012-09-22 11:09:41.000000000 +0200
@@ -36,6 +36,10 @@
 #include "TTYKeepAwake.h"
 #include "PMSettings.h"
 
+#ifndef nil
+#define nil NULL
+#endif
+
 #if !TARGET_OS_EMBEDDED
 
 
Seulement dans PowerManagement-158.10.9.pd1/PowerManagement.xcodeproj: kindred.mode1v3
Seulement dans PowerManagement-158.10.9.pd1/PowerManagement.xcodeproj: kindred.pbxuser
diff -ru PowerManagement-158.10.9/PowerManagement.xcodeproj/project.pbxproj PowerManagement-158.10.9.pd1/PowerManagement.xcodeproj/project.pbxproj
--- PowerManagement-158.10.9/PowerManagement.xcodeproj/project.pbxproj	2009-03-24 00:32:55.000000000 +0100
+++ PowerManagement-158.10.9.pd1/PowerManagement.xcodeproj/project.pbxproj	2012-09-22 11:10:13.000000000 +0200
@@ -373,7 +373,7 @@
 		72D0ECFE08F73FB600CCEA2F /* Info.plist */ = {isa = PBXFileReference; fileEncoding = 30; lastKnownFileType = text.plist.xml; name = Info.plist; path = AppleSmartBatteryManager/Info.plist; sourceTree = "<group>"; };
 		72D984480B20BE7800D66087 /* TTYKeepAwake.c */ = {isa = PBXFileReference; fileEncoding = 30; lastKnownFileType = sourcecode.c.c; path = TTYKeepAwake.c; sourceTree = "<group>"; };
 		72D984490B20BE7800D66087 /* TTYKeepAwake.h */ = {isa = PBXFileReference; fileEncoding = 30; lastKnownFileType = sourcecode.c.h; path = TTYKeepAwake.h; sourceTree = "<group>"; };
-		72E815720CFE470B00CF547E /* PowerManagement.bundle */ = {isa = PBXFileReference; includeInIndex = 0; lastKnownFileType = wrapper.cfbundle; path = PowerManagement.bundle; sourceTree = BUILT_PRODUCTS_DIR; };
+		72E815720CFE470B00CF547E /* PowerManagement.bundle */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = PowerManagement.bundle; sourceTree = BUILT_PRODUCTS_DIR; };
 		72E815730CFE470B00CF547E /* Power Management Bundle-Info copy.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist; path = "Power Management Bundle-Info copy.plist"; sourceTree = "<group>"; };
 		72FE21330A0179A300885E24 /* BatteryFaker.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = BatteryFaker.app; sourceTree = BUILT_PRODUCTS_DIR; };
 		72FE21400A0179F400885E24 /* BatteryFaker.kext */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = BatteryFaker.kext; sourceTree = BUILT_PRODUCTS_DIR; };
