diff -ru python-30.1.3/fix/sysconfig.py.ed python-30.1.3.pd1/fix/sysconfig.py.ed
--- python-30.1.3/fix/sysconfig.py.ed	2007-08-10 21:30:25.000000000 +0200
+++ python-30.1.3.pd1/fix/sysconfig.py.ed	2012-09-24 10:08:31.621902191 +0200
@@ -9,7 +9,7 @@
             if os.environ.has_key('ARCHFLAGS'):
                 arch = os.environ['ARCHFLAGS']
             else:
-                arch = '-arch i386 -arch ppc'
+                arch = '-arch i386'
             if arch.strip() != '':
                 cflags = cflags + ' ' + arch
                 ldshared = ldshared + ' ' + arch
diff -ru python-30.1.3/GNUmakefile python-30.1.3.pd1/GNUmakefile
--- python-30.1.3/GNUmakefile	2009-01-12 21:18:51.000000000 +0100
+++ python-30.1.3.pd1/GNUmakefile	2012-09-24 10:18:20.281890498 +0200
@@ -110,7 +110,6 @@
 	rm -rf $(PROJECT) && \
 	mv $(NAMEVERS) $(PROJECT) && \
 	ed - $(PROJECT)/configure.in < $(FIX)/configure.in.ed && \
-	ed - $(PROJECT)/Include/pymactoolbox.h < $(FIX)/pymactoolbox.h.ed && \
 	ed - $(PROJECT)/Lib/ctypes/__init__.py < $(FIX)/Lib_ctypes___init__.py.ed && \
 	ed - $(PROJECT)/Lib/xml/__init__.py < $(FIX)/Lib_xml___init__.py.ed && \
 	ed - $(PROJECT)/Lib/distutils/command/install.py < $(FIX)/install.py.ed && \
@@ -125,7 +124,6 @@
 	ed - $(PROJECT)/Mac/Makefile.in  < $(FIX)/Mac_Makefile.in.ed && \
 	ed - $(PROJECT)/Mac/Modules/cg/_CGmodule.c  < $(FIX)/_CGmodule.c.ed && \
 	ed - $(PROJECT)/Mac/Modules/launch/_Launchmodule.c < $(FIX)/_Launchmodule.c.ed && \
-	ed - $(PROJECT)/Makefile.pre.in < $(FIX)/Makefile.pre.in.ed && \
 	patch $(PROJECT)/Misc/python.man $(FIX)/python.man.patch && \
 	ed - $(PROJECT)/Modules/dlmodule.c < $(FIX)/dlmodule.c.ed && \
 	ed - $(PROJECT)/Modules/getpath.c < $(FIX)/getpath.c.ed && \
@@ -134,11 +132,6 @@
 	ed - $(PROJECT)/pyconfig.h.in < $(FIX)/pyconfig.h.in.ed && \
 	ed - $(PROJECT)/Python/ast.c < $(FIX)/ast.c.ed && \
 	ed - $(PROJECT)/Python/ceval.c < $(FIX)/ceval.c.ed && \
-	ed - $(PROJECT)/Python/mactoolboxglue.c < $(FIX)/mactoolboxglue.c.ed && \
-	ed - $(PROJECT)/setup.py < $(FIX)/setup.py.ed && \
-	for i in `find $(PROJECT)/Mac -name setup.py | xargs fgrep -l "'-framework', 'Carbon'"`; do \
-	    ed - $$i < $(FIX)/Mac_setup.py.ed || exit 1; \
-	done && \
 	for i in `find $(PROJECT)/Lib -name __init__.py -size 0c`; do \
 	    echo '#' > $$i; \
 	done
@@ -151,8 +144,6 @@
 	cd '$(OBJROOT)/$(PROJECT)' && patch -p3 -i $(FIX)/CVE-2008-3144-int-overflow.patch
 	cd '$(OBJROOT)/$(PROJECT)' && patch -p2 -i $(FIX)/CVE-2008-5031-expandtabs.patch
 	cd '$(OBJROOT)/$(PROJECT)' && patch -p3 -i $(FIX)/CESA-2008-008.patch
-	cd '$(OBJROOT)/$(PROJECT)' && autoconf
-	dtrace -h -s $(FIX)/pydtrace.d -o '$(PYDTRACE_H)'
 
 install: installpython installextras
 	install $(FIX)/audiotest.au $(DSTROOT)$(LIBPYTHONVERS)/email/test/data/audiotest.au
diff -ru python-30.1.3/Makefile python-30.1.3.pd1/Makefile
--- python-30.1.3/Makefile	2007-06-02 02:18:37.000000000 +0200
+++ python-30.1.3.pd1/Makefile	2012-09-24 10:08:44.233901940 +0200
@@ -3,7 +3,7 @@
 ##
 
 Project               = python
-Extra_Configure_Flags = --enable-ipv6 --with-threads --enable-framework=/System/Library/Frameworks --enable-toolbox-glue --enable-dtrace --with-system-ffi
+Extra_Configure_Flags = --enable-ipv6 --with-threads --enable-framework=/System/Library/Frameworks --with-system-ffi --disable-toolbox-glue --disable-dtrace
 
 ##---------------------------------------------------------------------
 # Extra_CC_Flags and Extra_LD_Flags are needed because CFLAGS gets overridden
@@ -13,7 +13,7 @@
 # Workaround for 3281234 (test_coercion failure due to non IEEE-754 in
 # optimizer): add -mno-fused-madd flag
 ##---------------------------------------------------------------------
-Extra_CC_Flags += -fno-common -Wno-long-double -mno-fused-madd -DENABLE_DTRACE
+Extra_CC_Flags += -fno-common -Wno-long-double -mno-fused-madd
 Extra_LD_Flags += -Wl,-F.
 Extra_Install_Flags   = DESTDIR='$(DSTROOT)'
 GnuAfterInstall       = fixup-after-install install-plist
