diff -ru python23-17.1.1/GNUmakefile python23-17.1.1.pd1/GNUmakefile
--- python23-17.1.1/GNUmakefile	2007-12-05 02:13:12.000000000 +0100
+++ python23-17.1.1.pd1/GNUmakefile	2012-09-24 11:26:38.881809083 +0200
@@ -64,7 +64,6 @@
 	ed - $(PROJECT)/Modules/getpath.c < $(FIX)/getpath.c.ed && \
 	ed - $(PROJECT)/Modules/_localemodule.c < $(FIX)/_localemodule.c.ed && \
 	ed - $(PROJECT)/pyconfig.h.in < $(FIX)/pyconfig.h.in.ed && \
-	ed - $(PROJECT)/Python/mactoolboxglue.c < $(FIX)/mactoolboxglue.c.ed
 	cd '$(OBJROOT)/$(PROJECT)' && patch -p1 -i $(FIX)/CVE-2007-4965-int-overflow.patch
 
 install: installpython
diff -ru python23-17.1.1/Makefile python23-17.1.1.pd1/Makefile
--- python23-17.1.1/Makefile	2006-09-06 09:00:15.000000000 +0200
+++ python23-17.1.1.pd1/Makefile	2012-09-24 11:26:36.257809135 +0200
@@ -3,7 +3,7 @@
 ##
 
 Project               = python
-Extra_Configure_Flags = --enable-ipv6 --with-threads --enable-framework=/System/Library/Frameworks --enable-toolbox-glue
+Extra_Configure_Flags = --enable-ipv6 --with-threads --enable-framework=/System/Library/Frameworks --disable-toolbox-glue
 
 ##---------------------------------------------------------------------
 # Extra_CC_Flags and Extra_LD_Flags are needed because CFLAGS gets overridden
