diff -ru ruby-67.6/Makefile ruby-67.6.pd1/Makefile
--- ruby-67.6/Makefile	2009-02-19 03:47:32.000000000 +0100
+++ ruby-67.6.pd1/Makefile	2012-09-24 14:46:19.629571096 +0200
@@ -65,15 +65,10 @@
 	find $(SYMROOT) -type d -empty -delete
 	$(STRIP) -x $(FW_VERSION_DIR)/usr/bin/ruby
 	$(MKDIR) $(ULB)
-	$(CP) $(FW_VERSION_DIR)/usr/bin/ruby $(ULB)/ruby64
-	lipo -remove ppc64 -output $(FW_VERSION_DIR)/usr/bin/ruby $(FW_VERSION_DIR)/usr/bin/ruby
-	lipo -remove x86_64 -output $(FW_VERSION_DIR)/usr/bin/ruby $(FW_VERSION_DIR)/usr/bin/ruby
 	$(STRIP) -x $(FW_VERSION_DIR)/usr/lib/ruby/$(VERSION)/$(SYSSTRING)/*.bundle
 	$(STRIP) -x $(FW_VERSION_DIR)/usr/lib/ruby/$(VERSION)/$(SYSSTRING)/*/*.bundle
 	$(RM) $(FW_VERSION_DIR)/usr/lib/libruby-static.a
 	$(STRIP) -x $(FW_VERSION_DIR)/usr/lib/libruby.$(MAJOR).dylib
-	ed - $(FW_VERSION_DIR)/usr/lib/ruby/$(VERSION)/$(SYSSTRING)/rbconfig.rb < $(SRCROOT)/patches/fix_rbconfig.ed
-	(cd $(FW_VERSION_DIR)/usr/lib/ruby/$(VERSION)/$(SYSSTRING) && sed -E -i '' "s/-arch (ppc|ppc64|i386|x86_64)//g" rbconfig.rb && patch -p0 < $(SRCROOT)/patches/rbconfig.diff)
 	$(LN) -fsh usr/lib/libruby.dylib $(FW_VERSION_DIR)/Ruby
 	(cd $(FW_VERSION_DIR) && for i in `find usr/lib/ruby/1.8/ -name "*.h"`; do $(LN) -fs ../$$i Headers/`basename $$i`; done)
 	$(MKDIR) $(DSTROOT)/usr/bin
@@ -126,11 +121,9 @@
 	ditto $(EXTRAS_DIR)/xcode_template $(TEMPL_DIR)
 	find $(TEMPL_DIR) -name ".svn" -print0 | xargs -0 rm -rf
 
-DTRACE_SAMPLE = $(DSTROOT)/Developer/Examples/Ruby/DTrace
 
 install-dtrace-sample:
-	$(MKDIR) $(DTRACE_SAMPLE)
-	$(CP) $(EXTRAS_DIR)/dtrace_sample/*.d $(DTRACE_SAMPLE)
+
 
 XRAY_PLUGINS = $(DSTROOT)/Developer/Library/Instruments/PlugIns
 
@@ -163,7 +156,6 @@
                  lib_rdoc_ri_ri_options.rb.diff \
                  lib_rdoc_ri_ri_paths.rb.diff \
                  lib_irb_init.rb.diff \
-                 dtrace.diff \
                  ruby_thread_hooks.diff \
                  rexml_bugs.diff \
 		 lib_rexml_entity.rb \
@@ -179,4 +171,3 @@
 		(cd $(SRCROOT)/$(Project) && patch -p0 < $(SRCROOT)/patches/$$patchfile) || exit 1; \
 	done
 	$(TOUCH) $(SRCROOT)/$(Project)/ext/win32ole/.document
-	dtrace -h -s $(EXTRAS_DIR)/dtrace.d -o $(SRCROOT)/$(Project)/dtrace.h
Seulement dans ruby-67.6.pd1: Makefile~
