diff -ru SQLite-46.1/Makefile SQLite-46.1.pd1/Makefile
--- SQLite-46.1/Makefile	2007-11-29 23:58:08.000000000 +0100
+++ SQLite-46.1.pd1/Makefile	2012-09-12 12:46:52.793536542 +0200
@@ -42,19 +42,19 @@
 	# remove static libraries and pkgconfig
 	${RM} -rf $(RC_Install_Prefix)/lib/*.*a $(RC_Install_Prefix)/lib/pkgconfig;
 	# fix the permissions on the tcl dylib
-	${CHMOD} 755 $(RC_Install_Prefix)/lib/sqlite3/libtclsqlite3.dylib; 
+	#${CHMOD} 755 $(RC_Install_Prefix)/lib/sqlite3/libtclsqlite3.dylib; 
 	# copy symbol-rich binary and libraries to SYMROOT
 	${CP} $(RC_Install_Prefix)/bin/sqlite3 $(SYMROOT)/sqlite3 
 	${CP} $(RC_Install_Prefix)/lib/libsqlite3.0.dylib $(SYMROOT)/libsqlite3.0.dylib 
-	${CP} $(RC_Install_Prefix)/lib/sqlite3/libtclsqlite3.dylib $(SYMROOT)/libtclsqlite3.dylib 
+	#${CP} $(RC_Install_Prefix)/lib/sqlite3/libtclsqlite3.dylib $(SYMROOT)/libtclsqlite3.dylib 
 	# create dSYM files
-	${DSYMUTIL} $(SYMROOT)/sqlite3
-	${DSYMUTIL} $(SYMROOT)/libsqlite3.0.dylib
-	${DSYMUTIL} $(SYMROOT)/libtclsqlite3.dylib
+	#${DSYMUTIL} $(SYMROOT)/sqlite3
+	#${DSYMUTIL} $(SYMROOT)/libsqlite3.0.dylib
+	#${DSYMUTIL} $(SYMROOT)/libtclsqlite3.dylib
 	# strip installed binaries
 	${STRIP} -S $(RC_Install_Prefix)/bin/sqlite3
 	${STRIP} -S $(RC_Install_Prefix)/lib/libsqlite3.0.dylib
-	${STRIP} -S $(RC_Install_Prefix)/lib/sqlite3/libtclsqlite3.dylib
+	#${STRIP} -S $(RC_Install_Prefix)/lib/sqlite3/libtclsqlite3.dylib
 	# setup man pages for installation
 	${MKDIR} $(RC_Install_Man)/man1; 
 	${CP} $(Sources)/sqlite3.1 $(RC_Install_Man)/man1/sqlite3.1; 
Seulement dans SQLite-46.1.pd1: Makefile~
