diff -ru subversion-16/Makefile subversion-16.pd1/Makefile
--- subversion-16/Makefile	2007-06-15 01:51:17.000000000 +0200
+++ subversion-16.pd1/Makefile	2012-09-24 02:31:37.914446739 +0200
@@ -5,10 +5,9 @@
                         post-install install-plist
 Extra_Configure_Flags = --enable-shared --disable-static \
                         --without-berkeley-db \
-                        --with-apxs --disable-mod-activation \
                         --with-apr=/usr --with-apr-util=/usr \
-                        --with-neon=/usr/local \
-                        --with-swig=/usr
+						--disable-keychain \
+                        --disable-swig
 
 include $(MAKEFILEPATH)/CoreOS/ReleaseControl/GNUSource.make
 
@@ -65,10 +64,6 @@
 		$(CP) $${file} $(SYMROOT); \
 		echo $(STRIP) -x $${file}; \
 		$(STRIP) -x $${file}; \
-		for arch in ppc64 x86_64; do \
-			echo lipo -remove $$arch -output $$file $$file; \
-			lipo -remove $$arch -output $$file $$file || true; \
-		done \
 	done
 	@for library in client delta diff fs fs_fs ra ra_dav ra_local ra_svn repos subr wc swig_perl swig_py swig_ruby; do \
 		$(CP) $(DSTROOT)/usr/lib/libsvn_$${library}-1.0.0.0.dylib $(SYMROOT); \
Seulement dans subversion-16.pd1: Makefile~
Seulement dans subversion-16.pd1/subversion/build: ltmain.sh~
