commit fa342f720d0e9a57e7fb775198d7f9ba685952bf
Author: William Kent <wjk011@gmail.com>
Date:   Tue Aug 27 13:59:02 2019 -0400

    Apply patch xcconfig

diff --git a/xcodeconfig/base.xcconfig b/xcodeconfig/base.xcconfig
index 74940e8..60f866d 100644
--- a/xcodeconfig/base.xcconfig
+++ b/xcodeconfig/base.xcconfig
@@ -1 +1,4 @@
-#include "<DEVELOPER_DIR>/Makefiles/CoreOS/Xcode/BSD.xcconfig"
+#include "../darwinbuild.xcconfig"
+
+GCC_WARN_ABOUT_DEPRECATED_FUNCTIONS = NO
+GCC_PREPROCESSOR_DEFINITIONS = __cold=
diff --git a/xcodeconfig/libasl.xcconfig b/xcodeconfig/libasl.xcconfig
index d55394e..8b9e313 100644
--- a/xcodeconfig/libasl.xcconfig
+++ b/xcodeconfig/libasl.xcconfig
@@ -29,10 +29,9 @@ LIBSYSCALL_LDFLAGS = -lsystem_kernel
 LIBSYSCALL_LDFLAGS[sdk=iphonesimulator*] = -lsystem_sim_kernel
 LIBM_LDFLAGS = -lsystem_m
 LIBDYLD_LDFLAGS = -ldyld
-LIBTRACE_LDFLAGS = -lsystem_trace
 LIBUNWIND_LDFLAGS = -lunwind
 
-OTHER_LDFLAGS = -all_load -L/usr/lib/system -umbrella System $(CR_LDFLAGS) $(LIBCOMPILER_RT_LDFLAGS) $(LIBDYLD_LDFLAGS) $(LIBSYSCALL_LDFLAGS) $(LIBPLATFORM_LDFLAGS) $(LIBPTHREAD_LDFLAGS) $(LIBMALLOC_LDFLAGS) $(LIBC_LDFLAGS) $(LIBUNWIND_LDFLAGS) $(LIBCLOSURE_LDFLAGS) -ldispatch $(LIBLAUNCH_LDFLAGS) -lxpc $(LIBTRACE_LDFLAGS) $(LIBNOTIFY_LDFLAGS) $(UPWARD_LINKS)
+OTHER_LDFLAGS = -all_load -L/usr/lib/system -umbrella System $(CR_LDFLAGS) $(LIBCOMPILER_RT_LDFLAGS) $(LIBDYLD_LDFLAGS) $(LIBSYSCALL_LDFLAGS) $(LIBPLATFORM_LDFLAGS) $(LIBPTHREAD_LDFLAGS) $(LIBMALLOC_LDFLAGS) $(LIBC_LDFLAGS) $(LIBUNWIND_LDFLAGS) $(LIBCLOSURE_LDFLAGS) -ldispatch $(LIBLAUNCH_LDFLAGS) -lxpc $(LIBNOTIFY_LDFLAGS) $(UPWARD_LINKS)
 
 UPWARD_LINKS = -Wl,-upward-lsystem_info
 
