diff -ru system_cmds-433.8/arch.tproj/Makefile system_cmds-433.8.pd1/arch.tproj/Makefile
--- system_cmds-433.8/arch.tproj/Makefile	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/arch.tproj/Makefile	2012-09-19 20:20:39.149263121 +0200
@@ -2,11 +2,12 @@
 Install_Dir = /usr/bin
 
 MFILES = arch.m
-Extra_Frameworks = -framework Foundation
+Extra_Frameworks = -framework Foundation -framework CoreFoundation
 
 MANPAGES = arch.1 machine.1
 
 Extra_CC_Flags = -DARCH_PROG=\"arch\" -DMACHINE_PROG=\"machine\"
+Extra_LD_Flags = -lobjc
 
 include $(MAKEFILEPATH)/CoreOS/ReleaseControl/BSDCommon.make
 
Seulement dans system_cmds-433.8.pd1/arch.tproj: Makefile~
diff -ru system_cmds-433.8/update.tproj/update.c system_cmds-433.8.pd1/update.tproj/update.c
--- system_cmds-433.8/update.tproj/update.c	2007-06-06 01:23:05.000000000 +0200
+++ system_cmds-433.8.pd1/update.tproj/update.c	2012-09-19 20:16:06.709268532 +0200
@@ -20,7 +20,7 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#include <sandbox.h>
+//#include <sandbox.h>
 #include <stdio.h>
 #include <unistd.h>
 
@@ -28,12 +28,14 @@
 main(void)
 {
 	const char* progname = "update";
+	/*
 	char* errmsg;
 	int res = sandbox_init(progname, SANDBOX_NAMED, &errmsg);
 	if (res) {
 		fprintf(stderr, "%s: sandbox_init: %s\n", progname, errmsg);
 		sandbox_free_error(errmsg);
 	}
+	*/
 
 	for (;;) {
 		sleep(30);
