diff -ru system_cmds-433.8/Makefile system_cmds-433.8.pd1/Makefile
--- system_cmds-433.8/Makefile	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/Makefile	2012-09-10 01:52:47.621748625 +0200
@@ -2,14 +2,14 @@
 
 Embedded=$(shell tconf --test TARGET_OS_EMBEDDED)
 
-SubProjects = dynamic_pager.tproj ac.tproj accton.tproj arch.tproj\
+SubProjects = dynamic_pager.tproj ac.tproj accton.tproj\
 	bootlog.tproj\
         dmesg.tproj\
         getconf.tproj getty.tproj hostinfo.tproj iostat.tproj\
         login.tproj makekey.tproj\
-        mkfile.tproj newgrp.tproj nvram.tproj passwd.tproj pwd_mkdb.tproj\
+        mkfile.tproj newgrp.tproj passwd.tproj nvram.tproj\
         reboot.tproj sync.tproj sysctl.tproj\
-        update.tproj vipw.tproj vifs.tproj zic.tproj zdump.tproj vm_stat.tproj\
+        vipw.tproj vifs.tproj zic.tproj zdump.tproj vm_stat.tproj\
         zprint.tproj latency.tproj sc_usage.tproj fs_usage.tproj\
         sadc.tproj sar.tproj sa.tproj \
 	dp_notify_lib nologin.tproj pagesize.tproj
@@ -17,7 +17,7 @@
 ifeq "$(Embedded)" "NO"
 SubProjects += at.tproj atrun.tproj\
 	auditd.tproj audit.tproj\
-	chkpasswd.tproj chpass.tproj\
+	chkpasswd.tproj\
 	dirhelper.tproj\
 	shutdown.tproj
 endif
Seulement dans system_cmds-433.8.pd1: Makefile~
