diff -ru system_cmds-433.8/passwd.tproj/Makefile system_cmds-433.8.pd1/passwd.tproj/Makefile
--- system_cmds-433.8/passwd.tproj/Makefile	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/passwd.tproj/Makefile	2012-09-11 01:33:19.892055563 +0200
@@ -7,7 +7,7 @@
 
 Embedded=$(shell tconf --test TARGET_OS_EMBEDDED)
 
-ifeq ($(Embedded),NO)
+ifeq ($(Embedded),YES)
 CFILES += nis_passwd.c od_passwd.c
 Extra_Frameworks = -F/System/Library/PrivateFrameworks -framework OpenDirectory -framework CoreFoundation
 Extra_CC_Flags = -F/System/Library/PrivateFrameworks
Seulement dans system_cmds-433.8.pd1/passwd.tproj: Makefile~
diff -ru system_cmds-433.8/passwd.tproj/passwd.c system_cmds-433.8.pd1/passwd.tproj/passwd.c
--- system_cmds-433.8/passwd.tproj/passwd.c	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/passwd.tproj/passwd.c	2012-09-11 01:43:17.028043704 +0200
@@ -23,7 +23,7 @@
 #include <TargetConditionals.h>
 
 #define INFO_FILE 1
-#if !TARGET_OS_EMBEDDED
+#if TARGET_OS_EMBEDDED
 #define INFO_NIS 2
 #define INFO_OPEN_DIRECTORY 3
 #endif
