diff -ru system_cmds-433.8/arch.tproj/Makefile system_cmds-433.8.pd1/arch.tproj/Makefile
--- system_cmds-433.8/arch.tproj/Makefile	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/arch.tproj/Makefile	2012-09-24 00:07:39.478618334 +0200
@@ -2,11 +2,12 @@
 Install_Dir = /usr/bin
 
 MFILES = arch.m
-Extra_Frameworks = -framework Foundation
+Extra_Frameworks = -framework Foundation -framework coreFoundation
 
 MANPAGES = arch.1 machine.1
 
 Extra_CC_Flags = -DARCH_PROG=\"arch\" -DMACHINE_PROG=\"machine\"
+Extra_LD_Flags = -lobjc
 
 include $(MAKEFILEPATH)/CoreOS/ReleaseControl/BSDCommon.make
 
diff -ru system_cmds-433.8/Makefile system_cmds-433.8.pd1/Makefile
--- system_cmds-433.8/Makefile	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/Makefile	2012-09-24 00:23:22.598599600 +0200
@@ -17,7 +17,7 @@
 ifeq "$(Embedded)" "NO"
 SubProjects += at.tproj atrun.tproj\
 	auditd.tproj audit.tproj\
-	chkpasswd.tproj chpass.tproj\
+    chpass.tproj\
 	dirhelper.tproj\
 	shutdown.tproj
 endif
diff -ru system_cmds-433.8/passwd.tproj/Makefile system_cmds-433.8.pd1/passwd.tproj/Makefile
--- system_cmds-433.8/passwd.tproj/Makefile	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/passwd.tproj/Makefile	2012-09-24 00:06:02.974620251 +0200
@@ -7,11 +7,11 @@
 
 Embedded=$(shell tconf --test TARGET_OS_EMBEDDED)
 
-ifeq ($(Embedded),NO)
-CFILES += nis_passwd.c od_passwd.c
-Extra_Frameworks = -F/System/Library/PrivateFrameworks -framework OpenDirectory -framework CoreFoundation
-Extra_CC_Flags = -F/System/Library/PrivateFrameworks
-endif
+#ifeq ($(Embedded),NO)
+#CFILES += nis_passwd.c od_passwd.c
+#Extra_Frameworks = -F/System/Library/PrivateFrameworks -#framework OpenDirectory -framework CoreFoundation
+#Extra_CC_Flags = -F/System/Library/PrivateFrameworks
+#endif
 
 include $(MAKEFILEPATH)/CoreOS/ReleaseControl/BSDCommon.make
 
Seulement dans system_cmds-433.8.pd1/passwd.tproj: Makefile~
diff -ru system_cmds-433.8/passwd.tproj/passwd.c system_cmds-433.8.pd1/passwd.tproj/passwd.c
--- system_cmds-433.8/passwd.tproj/passwd.c	2007-10-04 00:18:32.000000000 +0200
+++ system_cmds-433.8.pd1/passwd.tproj/passwd.c	2012-09-24 00:25:22.894597210 +0200
@@ -23,7 +23,8 @@
 #include <TargetConditionals.h>
 
 #define INFO_FILE 1
-#if !TARGET_OS_EMBEDDED
+//#if !TARGET_OS_EMBEDDED
+#if 0
 #define INFO_NIS 2
 #define INFO_OPEN_DIRECTORY 3
 #endif
diff -ru system_cmds-433.8/update.tproj/update.c system_cmds-433.8.pd1/update.tproj/update.c
--- system_cmds-433.8/update.tproj/update.c	2007-06-06 01:23:05.000000000 +0200
+++ system_cmds-433.8.pd1/update.tproj/update.c	2012-09-24 00:01:23.862625795 +0200
@@ -20,7 +20,7 @@
  * 
  * @APPLE_LICENSE_HEADER_END@
  */
-#include <sandbox.h>
+//#include <sandbox.h>
 #include <stdio.h>
 #include <unistd.h>
 
@@ -28,13 +28,14 @@
 main(void)
 {
 	const char* progname = "update";
+/*
 	char* errmsg;
 	int res = sandbox_init(progname, SANDBOX_NAMED, &errmsg);
 	if (res) {
 		fprintf(stderr, "%s: sandbox_init: %s\n", progname, errmsg);
 		sandbox_free_error(errmsg);
 	}
-
+*/
 	for (;;) {
 		sleep(30);
 		sync();
