diff --git a/makedefs/MakeInc.kernel b/makedefs/MakeInc.kernel
index 55de6d3..b0a63cf 100644
--- a/makedefs/MakeInc.kernel
+++ b/makedefs/MakeInc.kernel
@@ -345,6 +345,21 @@ do_installhdrs_mi:: $(DSTROOT)/$(KRESDIR)/Info.plist
 	$(_v)$(RM) $(DSTROOT)/$(KINCFRAME)/Resources
 	$(_v)$(LN) Versions/Current/Resources			\
 		   $(DSTROOT)/$(KINCFRAME)/Resources
+	$(_v)$(RM) $(DSTROOT)/$(SINCFRAME)/Versions/Current
+	$(_v)$(LN) $(SINCVERS) \
+		   $(DSTROOT)/$(SINCFRAME)/Versions/Current
+	$(_v)$(RM) $(DSTROOT)/$(SINCFRAME)/PrivateHeaders
+	$(_v)$(LN) Versions/Current/PrivateHeaders		\
+		   $(DSTROOT)/$(SINCFRAME)/PrivateHeaders
+	$(_v)$(RM) $(DSTROOT)/$(SINCFRAME)/Resources
+	$(_v)$(LN) Versions/Current/Resources			\
+		   $(DSTROOT)/$(SINCFRAME)/Resources
+	$(_v)$(RM) $(DSTROOT)/System/Library/Frameworks/IOKit.framework/Versions/Current
+	$(_v)$(LN) A $(DSTROOT)/System/Library/Frameworks/IOKit.framework/Versions/Current
+	$(_v)$(RM) $(DSTROOT)/System/Library/Frameworks/IOKit.framework/Headers
+	$(_v)$(LN) Versions/Current/Headers $(DSTROOT)/System/Library/Frameworks/IOKit.framework/Headers
+	$(_v)$(RM) $(DSTROOT)/System/Library/Frameworks/IOKit.framework/PrivateHeaders
+	$(_v)$(LN) Versions/Current/PrivateHeaders $(DSTROOT)/System/Library/Frameworks/IOKit.framework/PrivateHeaders
 
 $(DSTROOT)/$(KRESDIR)/Info.plist: $(SOURCE)/EXTERNAL_HEADERS/Info.plist
 	$(_v)$(MKDIR) $(DSTROOT)/$(KRESDIR)
