commit 11d7ad41aa1f2e400e7509ca2c08952dd949ed07
Author: William Kent <wjk011@gmail.com>
Date:   Sat Mar 21 16:09:08 2020 -0400

    Apply xnu_firehose_dir patch

diff --git a/makedefs/MakeInc.def b/makedefs/MakeInc.def
index a1030c3..4131b87 100644
--- a/makedefs/MakeInc.def
+++ b/makedefs/MakeInc.def
@@ -462,7 +462,7 @@ LDFLAGS_KERNEL_GEN = \
 	-Wl,-headerpad,152
 
 # LDFLAGS_KERNEL_SDK	= -L$(SDKROOT)/usr/local/lib/kernel -lfirehose_kernel
-LDFLAGS_KERNEL_SDK	= -L$(SDKROOT)/usr/local/lib/kernel
+LDFLAGS_KERNEL_SDK	= -L$(DEPROOT)/usr/local/lib/kernel
 
 LDFLAGS_KERNEL_RELEASE	=
 LDFLAGS_KERNEL_DEVELOPMENT     =
@@ -679,7 +679,7 @@ INCFLAGS_IMPORT	= $(patsubst %, -I$(OBJROOT)/EXPORT_HDRS/%, $(COMPONENT_IMPORT_L
 INCFLAGS_EXTERN	= -I$(SRCROOT)/EXTERNAL_HEADERS
 INCFLAGS_GEN	= -I$(SRCROOT)/$(COMPONENT) -I$(OBJROOT)/EXPORT_HDRS/$(COMPONENT)
 INCFLAGS_LOCAL	= -I.
-INCFLAGS_SDK	= -I$(SDKROOT)/usr/local/include/kernel
+INCFLAGS_SDK	= -I$(DEPROOT)/usr/local/include/kernel
 
 INCFLAGS	= $(INCFLAGS_LOCAL) $(INCFLAGS_GEN) $(INCFLAGS_IMPORT) $(INCFLAGS_EXTERN) $(INCFLAGS_MAKEFILE) $(INCFLAGS_SDK)
 
