diff --git a/makedefs/MakeInc.cmd b/makedefs/MakeInc.cmd
index e6dceaa..2708431 100644
--- a/makedefs/MakeInc.cmd
+++ b/makedefs/MakeInc.cmd
@@ -132,10 +132,10 @@ ifeq ($(DSYMUTIL),)
 	export DSYMUTIL := $(shell $(XCRUN) -sdk $(SDKROOT) -find dsymutil)
 endif
 ifeq ($(CTFCONVERT),)
-	export CTFCONVERT := $(shell $(XCRUN) -sdk $(SDKROOT) -find ctfconvert)
+	export CTFCONVERT := $(DEPROOT)/usr/local/bin/ctfconvert
 endif
 ifeq ($(CTFMERGE),)
-	export CTFMERGE :=  $(shell $(XCRUN) -sdk $(SDKROOT) -find ctfmerge)
+	export CTFMERGE :=  $(DEPROOT)/usr/local/bin/ctfmerge
 endif
 ifeq ($(CTFINSERT),)
 	export CTFINSERT := $(shell $(XCRUN) -sdk $(SDKROOT) -find ctf_insert)
