commit 81bd39f518ad4dd4a4416c59810e4abc1cdc7a5e
Author: William Kent <wjk011@gmail.com>
Date:   Sat Mar 21 16:19:03 2020 -0400

    Apply remove_cfg_adr patch

diff --git a/osfmk/i386/cpuid.h b/osfmk/i386/cpuid.h
index c803080..4d41bbb 100644
--- a/osfmk/i386/cpuid.h
+++ b/osfmk/i386/cpuid.h
@@ -492,17 +492,6 @@ is_xeon_sp(uint8_t platid)
 	if (platid == PLATID_XEON_SP_1 || platid == PLATID_XEON_SP_2) {
 		return 1;
 	}
-	if (platid != PLATID_MAYBE_XEON_SP) {
-		return 0;
-	}
-	boolean_t intrs = ml_set_interrupts_enabled(FALSE);
-	outl(cfgAdr, XeonCapID5);
-	uint32_t cap5reg = inl(cfgDat);
-	ml_set_interrupts_enabled(intrs);
-	/* Read from PCI config space 1:30:3:0x98 [bits 13:9] */
-	if (bitfield32(cap5reg, 13, 9) == 3) {
-		return 1;
-	}
 	return 0;
 }
 
diff --git a/osfmk/i386/hpet.c b/osfmk/i386/hpet.c
index 8b9f03b..6a997b2 100644
--- a/osfmk/i386/hpet.c
+++ b/osfmk/i386/hpet.c
@@ -189,13 +189,6 @@ hpet_request(uint32_t cpu)
 static void
 map_rcbaArea(void)
 {
-	/*
-	 * Get RCBA area physical address and map it
-	 */
-	outl(cfgAdr, lpcCfg | (0xF0 & 0xFC));
-	rcbaAreap = inl(cfgDat | (0xF0 & 0x03));
-	rcbaArea = io_map_spec(rcbaAreap & -4096, PAGE_SIZE * 4, VM_WIMG_IO);
-	kprintf("RCBA: vaddr = %lX, paddr = %08X\n", (unsigned long)rcbaArea, rcbaAreap);
 }
 
 /*
