commit e8eabb0a5331659c3972a67947d0dee1adc135d3
Author: William Kent <wjk011@gmail.com>
Date:   Sat Mar 21 16:26:17 2020 -0400

    Apply ipsec_nexus

diff --git a/bsd/net/if_ipsec.c b/bsd/net/if_ipsec.c
index e967cad..17aba18 100644
--- a/bsd/net/if_ipsec.c
+++ b/bsd/net/if_ipsec.c
@@ -2706,11 +2706,13 @@ ipsec_ctl_connect(kern_ctl_ref kctlref,
 	 * Mark the data path as ready.
 	 * If kpipe nexus is being used then the data path is marked ready only when a kpipe channel is connected.
 	 */
+#if IPSEC_NEXUS
 	if (pcb->ipsec_kpipe_count == 0) {
 		lck_mtx_lock(&pcb->ipsec_pcb_data_move_lock);
 		IPSEC_SET_DATA_PATH_READY(pcb);
 		lck_mtx_unlock(&pcb->ipsec_pcb_data_move_lock);
 	}
+#endif
 
 	/* The interfaces resoures allocated, mark it as running */
 	ifnet_set_flags(pcb->ipsec_ifp, IFF_RUNNING, IFF_RUNNING);
