diff --git a/SETUP/config/Makefile b/SETUP/config/Makefile
index fb79f3f..db44d6f 100644
--- a/SETUP/config/Makefile
+++ b/SETUP/config/Makefile
@@ -20,7 +20,7 @@ config: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
diff --git a/SETUP/decomment/Makefile b/SETUP/decomment/Makefile
index a22212f..0dffd4d 100644
--- a/SETUP/decomment/Makefile
+++ b/SETUP/decomment/Makefile
@@ -18,7 +18,7 @@ decomment: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
diff --git a/SETUP/installfile/Makefile b/SETUP/installfile/Makefile
index 4ad7a74..9a43a43 100644
--- a/SETUP/installfile/Makefile
+++ b/SETUP/installfile/Makefile
@@ -18,7 +18,7 @@ installfile: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
diff --git a/SETUP/json_compilation_db/Makefile b/SETUP/json_compilation_db/Makefile
index 18af26b..1504915 100644
--- a/SETUP/json_compilation_db/Makefile
+++ b/SETUP/json_compilation_db/Makefile
@@ -18,7 +18,7 @@ json_compilation_db: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
diff --git a/SETUP/kextsymboltool/Makefile b/SETUP/kextsymboltool/Makefile
index dde295b..026164a 100644
--- a/SETUP/kextsymboltool/Makefile
+++ b/SETUP/kextsymboltool/Makefile
@@ -18,7 +18,7 @@ kextsymboltool: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
diff --git a/SETUP/replacecontents/Makefile b/SETUP/replacecontents/Makefile
index 45459e4..6db0bb9 100644
--- a/SETUP/replacecontents/Makefile
+++ b/SETUP/replacecontents/Makefile
@@ -18,7 +18,7 @@ replacecontents: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
diff --git a/SETUP/setsegname/Makefile b/SETUP/setsegname/Makefile
index 70a55a7..25d5b7c 100644
--- a/SETUP/setsegname/Makefile
+++ b/SETUP/setsegname/Makefile
@@ -18,7 +18,7 @@ setsegname: $(OBJS)
 	$(call makelog,$(ColorH)HOST_LD$(Color0)       $(ColorF)$@$(Color0))
 	$(_v)$(HOST_CC) $(LDFLAGS) -o $@ $^
 	$(call makelog,$(ColorH)HOST_CODESIGN$(Color0) $(ColorF)$@$(Color0))
-	$(_v)env CODESIGN_ALLOCATE=$(HOST_CODESIGN_ALLOCATE) $(HOST_CODESIGN) -s - $@
+	$(_v)$(HOST_CODESIGN) -s - $@
 
 %.o: %.c
 	$(call makelog,$(ColorH)HOST_CC$(Color0)       $(ColorF)$@$(Color0))
