diff -ru xnu-1228.15.4/makedefs/MakeInc.def xnu-1228.15.4.pd1/makedefs/MakeInc.def
--- xnu-1228.15.4/makedefs/MakeInc.def	2009-08-07 13:02:46.000000000 -0700
+++ xnu-1228.15.4.pd1/makedefs/MakeInc.def	2012-09-08 13:08:19.000000000 -0700
@@ -36,7 +36,7 @@
 # Architecture options
 #
 ifndef SUPPORTED_ARCH_CONFIGS
-export SUPPORTED_ARCH_CONFIGS = PPC I386 ARM
+export SUPPORTED_ARCH_CONFIGS = I386
 endif
 
 ifndef ARCH_CONFIGS
@@ -142,7 +142,7 @@
 export INSTALL_ARCHS_LC := $(shell printf "%s" "$(INSTALL_ARCHS)" | $(TR) A-Z a-z)
 endif
 
-export INSTALL_ARCH_DEFAULT	= PPC
+export INSTALL_ARCH_DEFAULT	= I386
 
 #
 # Standard defines list
diff -ru xnu-1228.15.4/makedefs/MakeInc.rule xnu-1228.15.4.pd1/makedefs/MakeInc.rule
--- xnu-1228.15.4/makedefs/MakeInc.rule	2009-08-07 13:02:48.000000000 -0700
+++ xnu-1228.15.4.pd1/makedefs/MakeInc.rule	2012-09-08 14:49:27.000000000 -0700
@@ -611,10 +611,6 @@
 	$(_v)${KCC} -c ${filter-out ${${join $@,_CFLAGS_RM}}, ${CFLAGS}} ${${join $@,_CFLAGS_ADD}} ${INCFLAGS} ${${join $@,_INCFLAGS}} $(OBJPATH)/version.c -o $(OBJPATH)/version.o
 	@echo LD mach_kernel.sys
 	$(_v)$(LD) $(LDFLAGS_KERNEL) $(addprefix $(TARGET)/,$(foreach component,$(COMPONENT_LIST), $(addprefix $(component)/$(firstword $($(addsuffix _KERNEL_CONFIG, $(shell printf $(component) | tr a-z A-Z))) $(KERNEL_CONFIG))/, $(addsuffix .o, $(component))))) $(OBJPATH)/version.o -o $(TARGET)/mach_kernel.sys $(LD_KERNEL_LIBS)
-	@echo DSYMUTIL mach_kernel.sys
-	$(_v)if [ $(BUILD_DWARF)  -eq  1 ]; then \
-		$(DSYMUTIL) $(DSYMUTIL_FLAGS) $(TARGET)/mach_kernel.sys -o $(TARGET)/mach_kernel.sys.dSYM > /dev/null; \
-	fi;
 	@echo STRIP mach_kernel
 	$(_v)$(STRIP) $(STRIP_FLAGS) $(TARGET)/mach_kernel.sys -o $(TARGET)/mach_kernel
 
@@ -672,14 +668,7 @@
 	fi;							\
 	if [ "`echo $(INSTALL_ARCHS_LC) | wc -w`" -eq 1 ]; then	\
 		$(RM) $(RMFLAGS) $@;				\
-		$(INSTALL) $(FILE_INSTALL_FLAGS) $< $@;		\
-		if [ $(BUILD_DWARF) -eq 1 ]; then			\
-			$(RM) -rf $@.dSYM;				\
-			$(MKDIR) -p -m 0755 $@.dSYM/$(DSYMBUILDDIR);	\
-			$(INSTALL) $(INSTALL_FLAGS)			\
-				$<.dSYM/$(DSYMBUILDDIR)/$(notdir $<)	\
-				$@.dSYM/$(DSYMBUILDDIR)/$(notdir $@);	\
-		fi;							\
+		$(INSTALL) $(FILE_INSTALL_FLAGS) $< $@;		\						\
 	else							\
 		if [ ! -e $@ ]; then				\
 			echo >empty_file_$(notdir $@);			\
@@ -758,16 +747,6 @@
 	if [ "`echo $(INSTALL_ARCHS_LC) | wc -w`" -eq 1 ]; then	\
 		$(RM) $(RMFLAGS) $@;				\
 		$(INSTALL) $(INSTALL_FLAGS) $< $@;			\
-		if [ $(BUILD_DWARF) -eq 1 ]; then			\
-			$(DSYMUTIL) $(DSYMUTIL_FLAGS)			\
-				$(TARGET)/mach_kernel.sys		\
-				-o $(TARGET)/mach_kernel.sys.dSYM;	\
-			$(RM) -rf $@.dSYM;				\
-			$(MKDIR) -p -m 0755 $@.dSYM/$(DSYMBUILDDIR);	\
-			$(INSTALL) $(INSTALL_FLAGS)			\
-				$<.dSYM/$(DSYMBUILDDIR)/$(notdir $<)	\
-				$@.dSYM/$(DSYMBUILDDIR)/$(notdir $@);	\
-		fi;							\
 	else							\
 		if [ ! -e $@ ]; then				\
 			echo >empty_filesys_$(notdir $@);	\
@@ -776,25 +755,6 @@
 			$(RM) $(RMFLAGS) empty_filesys_$(notdir $@);	\
 		fi;							\
 		$(LIPO) $@ -replace $(ARCH_CONFIG_LC)  $< -o $@;	\
-									\
-		if [ $(BUILD_DWARF) -eq 1 ]; then			       \
-			if [ ! -e $@.dSYM/$(DSYMBUILDDIR)/$(notdir $@) ]; then \
-				echo >empty_filesys_$(notdir $@);	       \
-				lipo_arg="$(subst _empty_file, empty_filesys_$(notdir $@),$(foreach lipo_arch,$(INSTALL_ARCHS_LC), $(addprefix -arch , $(addsuffix _empty_file, $(lipo_arch)))))"; \
-				$(MKDIR) -p -m 0755 $@.dSYM/$(DSYMBUILDDIR);   \
-				$(LIPO) $${lipo_arg} -create		       \
-					-output				       \
-					$@.dSYM/$(DSYMBUILDDIR)/$(notdir $@);  \
-				$(RM) $(RMFLAGS) empty_filesys_$(notdir $@);   \
-			fi;						       \
-			$(DSYMUTIL) $(DSYMUTIL_FLAGS)			\
-				$(TARGET)/mach_kernel.sys		\
-				-o $(TARGET)/mach_kernel.sys.dSYM;	\
-			$(LIPO) $@.dSYM/$(DSYMBUILDDIR)/$(notdir $@)	       \
-				-replace $(ARCH_CONFIG_LC)		       \
-				$<.dSYM/$(DSYMBUILDDIR)/$(notdir $<)	       \
-				-o $@.dSYM/$(DSYMBUILDDIR)/$(notdir $@);       \
-		fi;							       \
 	fi
 	$(INSTALL) $(INSTALL_FLAGS) $(SOURCE)kgmacros $(SYMROOT)$(INSTALL_FILE_DIR)
 
