commit 2575332ab9a32fcd9a6fcfc6b2befc78cbe0d6f4
Author: William Kent <wjk011@gmail.com>
Date:   Sat Mar 21 16:14:37 2020 -0400

    Apply no_uncrustify patch

diff --git a/Makefile b/Makefile
index 64822cd..c833815 100644
--- a/Makefile
+++ b/Makefile
@@ -165,8 +165,7 @@ TOP_TARGETS =								\
 	install install_desktop install_embedded			\
 	install_release_embedded install_development_embedded		\
 	installopensource						\
-	cscope tags TAGS checkstyle restyle check_uncrustify uncrustify				\
-	help
+	cscope tags TAGS help
 
 DEFAULT_TARGET = all
 
diff --git a/makedefs/MakeInc.top b/makedefs/MakeInc.top
index dd1070e..bbe4661 100644
--- a/makedefs/MakeInc.top
+++ b/makedefs/MakeInc.top
@@ -563,7 +563,7 @@ else ifeq ($(RC_ProjectName),xnu_headers_driverkit)
 install: installhdrs_desktop
 else
 
-install: installhdrs install_textfiles install_config install_kernels install_aliases checkstyle
+install: installhdrs install_textfiles install_config install_kernels install_aliases
 endif
 
 .PHONY: install_embedded install_release_embedded install_development_embedded install_desktop
@@ -680,21 +680,6 @@ TAGS: cscope.files
 	@-cat cscope.files | etags -l auto -S - 2> /dev/null
 	@rm -f cscope.files 2> /dev/null
 
-#
-# Check or reformat source code for official xnu code style
-#
-.PHONY: checkstyle restyle check_uncrustify uncrustify
-
-# User-friendly aliases for those who prefer to remember the name of the tool.
-check_uncrustify: checkstyle
-uncrustify: restyle
-
-checkstyle:
-	${_V}$(SRCROOT)/tools/uncrustify.sh
-
-restyle:
-	${_V}$(SRCROOT)/tools/uncrustify.sh -f
-
 .PHONY: help
 
 help:
