diff --git a/bsd/sys/make_symbol_aliasing.sh b/bsd/sys/make_symbol_aliasing.sh
index 8e98dee..ae4124c 100755
--- a/bsd/sys/make_symbol_aliasing.sh
+++ b/bsd/sys/make_symbol_aliasing.sh
@@ -34,7 +34,7 @@ fi
 SDKROOT="$1"
 OUTPUT="$2"
 
-AVAILABILITY_PL="${SDKROOT}/${DRIVERKITROOT}/usr/local/libexec/availability.pl"
+AVAILABILITY_PL="${DEPROOT}/usr/local/libexec/availability.pl"
 
 if [ ! -x "${AVAILABILITY_PL}" ] ; then
     echo "Unable to locate ${AVAILABILITY_PL} (or not executable)" >&2
