diff --git a/bsd/sys/make_symbol_aliasing.sh b/bsd/sys/make_symbol_aliasing.sh
index ef47e37..a1e694a 100755
--- a/bsd/sys/make_symbol_aliasing.sh
+++ b/bsd/sys/make_symbol_aliasing.sh
@@ -34,8 +34,8 @@ fi
 SDKROOT="$1"
 OUTPUT="$2"

-if [ ! -x "${SDKROOT}/usr/local/libexec/availability.pl" ] ; then
-    echo "Unable to locate ${SDKROOT}/usr/local/libexec/availability.pl (or not executable)" >&2
+if [ ! -x "${DEPENDENCIES_DIR}/usr/local/libexec/availability.pl" ] ; then
+    echo "Unable to locate ${DEPENDENCIES_DIR}/usr/local/libexec/availability.pl (or not executable)" >&2
     exit 1
 fi

@@ -74,7 +74,7 @@ cat <<EOF

 EOF

-for ver in $(${SDKROOT}/usr/local/libexec/availability.pl --ios) ; do
+for ver in $(${DEPENDENCIES_DIR}/usr/local/libexec/availability.pl --ios) ; do
     ver_major=${ver%.*}
     ver_minor=${ver#*.}
     value=$(printf "%d%02d00" ${ver_major} ${ver_minor})
@@ -87,7 +87,7 @@ for ver in $(${SDKROOT}/usr/local/libexec/availability.pl --ios) ; do
     echo ""
 done

-for ver in $(${SDKROOT}/usr/local/libexec/availability.pl --macosx) ; do
+for ver in $(${DEPENDENCIES_DIR}/usr/local/libexec/availability.pl --macosx) ; do
     set -- $(echo "$ver" | tr '.' ' ')
     ver_major=$1
     ver_minor=$2
