diff --git a/config/MASTER b/config/MASTER
index f0900b3..a0bcd3e 100644
--- a/config/MASTER
+++ b/config/MASTER
@@ -146,9 +146,9 @@ options		CONFIG_EXT_RESOLVER # e.g. memberd		# <config_ext_resolver>
 options		CONFIG_SEARCHFS	# searchfs syscall support	# <config_searchfs>
 options		CONFIG_MNT_SUID # allow suid binaries  # <config_mnt_suid>
 options		CONFIG_MNT_ROOTSNAP # allow rooting from snapshot # <config_mnt_rootsnap>
-options 	CONFIG_ROSV_STARTUP # allow read-only system volume startup # <config_rosv_startup>
+#options 	CONFIG_ROSV_STARTUP # allow read-only system volume startup # <config_rosv_startup>
 options		CONFIG_FIRMLINKS # support "firmlinks" # <config_firmlinks>
-options 	CONFIG_MOUNT_VM # mount VM volume on startup # <config_mount_vm>
+#options 	CONFIG_MOUNT_VM # mount VM volume on startup # <config_mount_vm>
 options		CONFIG_DATALESS_FILES # support dataless file materialization # <config_dataless_files>
 
 #
