commit d54be66490a429e4dace85d320478f3f64f9c3f0
Author: William Kent <wjk011@gmail.com>
Date:   Sat Mar 21 16:29:20 2020 -0400

    Apply fix_dlil patch

diff --git a/bsd/net/dlil.c b/bsd/net/dlil.c
index 7a11991..76c997e 100644
--- a/bsd/net/dlil.c
+++ b/bsd/net/dlil.c
@@ -131,6 +131,9 @@
 #include <net/necp.h>
 #endif /* NECP */
 
+// From if_vlan_var.h
+#define EVL_VLID_MASK   0x0FFF
+#define EVL_VLANOFTAG(tag) ((tag) & EVL_VLID_MASK)
 
 #include <os/log.h>
 
