commit b316011b4f2b3ee5ee5f04f16e97addd0694cb4f
Author: William Kent <wjk011@gmail.com>
Date:   Wed Apr 29 17:05:34 2020 -0400

    Apply patch fix_fork_asm

diff --git a/libsyscall/custom/__fork.s b/libsyscall/custom/__fork.s
index ffcbe5a..7870b66 100644
--- a/libsyscall/custom/__fork.s
+++ b/libsyscall/custom/__fork.s
@@ -93,7 +93,7 @@ L1:
 	
 	//child here...
 	xorq	%rax, %rax
-	PICIFY(__current_pid)
+	movq	__current_pid(%rip), %r11
 	movl	%eax,(%r11)
 L2:
 	// parent ends up here skipping child portion
diff --git a/libsyscall/custom/__ptrace.s b/libsyscall/custom/__ptrace.s
index 607a26e..b7b8504 100644
--- a/libsyscall/custom/__ptrace.s
+++ b/libsyscall/custom/__ptrace.s
@@ -45,7 +45,7 @@ LEAF(___ptrace, 0)
 
 LEAF(___ptrace, 0)
 	xorq	%rax,%rax
-	PICIFY(_errno)
+	movq	_errno(%rip), %r11
 	movl	%eax,(%r11)
 	UNIX_SYSCALL_NONAME(ptrace, 4, cerror)
 	ret
