diff --git a/src/llvm/include/llvm/TextAPI/MachO/Architecture.def b/src/llvm/include/llvm/TextAPI/MachO/Architecture.def
index 2fcae3b2..d5175f8c 100644
--- a/src/llvm/include/llvm/TextAPI/MachO/Architecture.def
+++ b/src/llvm/include/llvm/TextAPI/MachO/Architecture.def
@@ -37,3 +37,7 @@ ARCHINFO(armv7em, MachO::CPU_TYPE_ARM, MachO::CPU_SUBTYPE_ARM_V7EM, 32)
 ///
 ARCHINFO(arm64, MachO::CPU_TYPE_ARM64, MachO::CPU_SUBTYPE_ARM64_ALL, 64)
 ARCHINFO(arm64e, MachO::CPU_TYPE_ARM64, MachO::CPU_SUBTYPE_ARM64E, 64)
+
+// riscv, cpu type 24 as ld64 defines it
+ARCHINFO(riscv32, 24, 0, 32)
+ARCHINFO(riscv64, 24 | MachO::CPU_ARCH_ABI64, 0, 64)
